*** ./smali/android/provider/Settings$System.smali	2013-11-10 11:53:26.801672875 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/android/provider/Settings$System.smali	2013-11-10 11:54:46.537674917 +0800
***************
*** 138,143 ****
--- 138,145 ----
  
  .field public static final DTMF_TONE_WHEN_DIALING:Ljava/lang/String; = "dtmf_tone"
  
+ .field public static final EMERGENCY_TONE:Ljava/lang/String; = "emergency_tone"
+ 
  .field public static final END_BUTTON_BEHAVIOR:Ljava/lang/String; = "end_button_behavior"
  
  .field public static final END_BUTTON_BEHAVIOR_DEFAULT:I = 0x2
***************
*** 2425,2430 ****
--- 2427,2436 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     invoke-static {p0, p1, p2, p3}, Landroid/provider/Settings$Injector;->putSecureStringForUser(Landroid/content/ContentResolver;Ljava/lang/String;Ljava/lang/String;I)Z
+ 
+     move-result v0
+ 
      :goto_0
      return v0
  
***************
*** 2479,2484 ****
--- 2485,2493 ----
      move-result-object v2
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
+     invoke-static {p0, p1, p2, p3}, Landroid/provider/Settings$Injector;->putGlobalStringForUser(Landroid/content/ContentResolver;Ljava/lang/String;Ljava/lang/String;I)Z
+ 
+     move-result v0
  
      goto :goto_0
  
