*** ./smali/com/android/server/power/ShutdownThread.smali	2013-11-10 11:52:19.529671153 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/services.jar.out/./smali/com/android/server/power/ShutdownThread.smali	2013-11-10 11:53:38.933673186 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/power/ShutdownThread$CloseDialogReceiver;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/power/ShutdownThread$CloseDialogReceiver;,
!         Lcom/android/server/power/ShutdownThread$Injector;
      }
  .end annotation
  
***************
*** 346,351 ****
--- 347,414 ----
      goto :goto_2
  .end method
  
+ .method static getIsStarted()Z
+     .locals 1
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/server/power/ShutdownThread;->sIsStarted:Z
+ 
+     return v0
+ .end method
+ 
+ .method static getIsStartedGuard()Ljava/lang/Object;
+     .locals 1
+ 
+     .prologue
+     sget-object v0, Lcom/android/server/power/ShutdownThread;->sIsStartedGuard:Ljava/lang/Object;
+ 
+     return-object v0
+ .end method
+ 
+ .method static getReboot()Z
+     .locals 1
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/server/power/ShutdownThread;->mReboot:Z
+ 
+     return v0
+ .end method
+ 
+ .method private static getResourceId(I)I
+     .locals 1
+     .parameter "id"
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/server/power/ShutdownThread;->mReboot:Z
+ 
+     if-eqz v0, :cond_0
+ 
+     const p0, 0x60c019a
+ 
+     .end local p0
+     :cond_0
+     return p0
+ .end method
+ 
+ .method private static getTitleResourceId()I
+     .locals 1
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/server/power/ShutdownThread;->mReboot:Z
+ 
+     if-eqz v0, :cond_0
+ 
+     const v0, 0x60c000d
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const v0, 0x60c0026
+ 
+     goto :goto_0
+ .end method
+ 
  .method public static reboot(Landroid/content/Context;Ljava/lang/String;Z)V
      .locals 1
      .parameter "context"
***************
*** 478,483 ****
--- 541,548 ----
  
      sput-boolean v0, Lcom/android/server/power/ShutdownThread;->mReboot:Z
  
+     const/4 v0, 0x0
+ 
      sput-boolean v0, Lcom/android/server/power/ShutdownThread;->mRebootSafeMode:Z
  
      const/4 v0, 0x0
***************
*** 489,494 ****
--- 554,579 ----
      return-void
  .end method
  
+ .method static setReboot(Z)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     sput-boolean p0, Lcom/android/server/power/ShutdownThread;->mReboot:Z
+ 
+     return-void
+ .end method
+ 
+ .method static setRebootReason(Ljava/lang/String;)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     sput-object p0, Lcom/android/server/power/ShutdownThread;->mRebootReason:Ljava/lang/String;
+ 
+     return-void
+ .end method
+ 
  .method public static shutdown(Landroid/content/Context;Z)V
      .locals 1
      .parameter "context"
***************
*** 556,561 ****
--- 641,650 ----
  
      .local v2, resourceId:I
      :goto_1
+     invoke-static {v2}, Lcom/android/server/power/ShutdownThread$Injector;->getResourceId(I)I
+ 
+     move-result v2
+ 
      const-string v3, "ShutdownThread"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 639,644 ****
--- 728,740 ----
  
      sget-object v3, Lcom/android/server/power/ShutdownThread;->sConfirmDialog:Landroid/app/AlertDialog;
  
+ 
+     sget-object v3, Lcom/android/server/power/ShutdownThread;->sConfirmDialog:Landroid/app/AlertDialog;
+ 
+     invoke-static {v3}, Lcom/android/server/power/ShutdownThread$Injector;->setDialogTitle(Landroid/app/Dialog;)V
+ 
+     sget-object v3, Lcom/android/server/power/ShutdownThread;->sConfirmDialog:Landroid/app/AlertDialog;
+ 
      iput-object v3, v0, Lcom/android/server/power/ShutdownThread$CloseDialogReceiver;->dialog:Landroid/app/Dialog;
  
      sget-object v3, Lcom/android/server/power/ShutdownThread;->sConfirmDialog:Landroid/app/AlertDialog;
