*** ./smali/android/content/pm/ApplicationInfo.smali	2013-11-08 15:41:20.512921506 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/content/pm/ApplicationInfo.smali	2013-11-08 15:38:25.864921528 +0800
***************
*** 26,31 ****
--- 26,33 ----
      .end annotation
  .end field
  
+ .field public static final FLAG_ACCESS_CONTROL_PASSWORD:I = -0x80000000
+ 
  .field public static final FLAG_ALLOW_BACKUP:I = 0x8000
  
  .field public static final FLAG_ALLOW_CLEAR_USER_DATA:I = 0x40
***************
*** 36,41 ****
--- 38,45 ----
  
  .field public static final FLAG_DEBUGGABLE:I = 0x2
  
+ .field public static final FLAG_DISABLE_AUTOSTART:I = 0x40000000
+ 
  .field public static final FLAG_EXTERNAL_STORAGE:I = 0x40000
  
  .field public static final FLAG_FACTORY_TEST:I = 0x10
***************
*** 58,63 ****
--- 62,69 ----
  
  .field public static final FLAG_RESTORE_ANY_VERSION:I = 0x20000
  
+ .field public static final FLAG_SHOW_FLOATING_WINDOW:I = 0x8000000
+ 
  .field public static final FLAG_STOPPED:I = 0x200000
  
  .field public static final FLAG_SUPPORTS_LARGE_SCREENS:I = 0x800
