*** ./smali/android/content/pm/PackageParser.smali	2013-11-18 12:21:46.762342850 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/content/pm/PackageParser.smali	2013-11-18 12:18:50.466339541 +0800
***************
*** 21,27 ****
          Landroid/content/pm/PackageParser$ParseComponentArgs;,
          Landroid/content/pm/PackageParser$ParsePackageItemArgs;,
          Landroid/content/pm/PackageParser$SplitPermissionInfo;,
!         Landroid/content/pm/PackageParser$NewPermissionInfo;
      }
  .end annotation
  
--- 21,28 ----
          Landroid/content/pm/PackageParser$ParseComponentArgs;,
          Landroid/content/pm/PackageParser$ParsePackageItemArgs;,
          Landroid/content/pm/PackageParser$SplitPermissionInfo;,
!         Landroid/content/pm/PackageParser$NewPermissionInfo;,
!         Landroid/content/pm/PackageParser$Injector;
      }
  .end annotation
  
***************
*** 6509,6517 ****
      move-result v7
  
      .local v7, priority:I
      move-object/from16 v0, p5
  
!     invoke-virtual {v0, v7}, Landroid/content/pm/PackageParser$IntentInfo;->setPriority(I)V
  
      const/4 v13, 0x0
  
--- 6510,6524 ----
      move-result v7
  
      .local v7, priority:I
+     move/from16 v0, p4
+ 
+     invoke-static {v0, v7}, Landroid/content/pm/PackageParser$Injector;->checkPriority(II)I
+ 
+     move-result v13
+ 
      move-object/from16 v0, p5
  
!     invoke-virtual {v0, v13}, Landroid/content/pm/PackageParser$IntentInfo;->setPriority(I)V
  
      const/4 v13, 0x0
  
***************
*** 7357,7362 ****
--- 7364,7375 ----
  
      invoke-static {v0, v3}, Landroid/content/pm/PackageParser;->validateName(Ljava/lang/String;Z)Ljava/lang/String;
  
+     move-result-object v3
+ 
+     move-object/from16 v0, v34
+ 
+     invoke-static {v0, v3}, Landroid/content/pm/PackageParser$Injector;->filterNameError(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v29
  
      .local v29, nameError:Ljava/lang/String;
***************
*** 9513,9518 ****
--- 9526,9535 ----
  
      invoke-static {v6, v12}, Landroid/content/pm/PackageParser;->validateName(Ljava/lang/String;Z)Ljava/lang/String;
  
+     move-result-object v12
+ 
+     invoke-static {v6, v12}, Landroid/content/pm/PackageParser$Injector;->filterNameError(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v4
  
      .local v4, nameError:Ljava/lang/String;
***************
*** 10137,10142 ****
--- 10154,10163 ----
      :cond_5
      invoke-static {v1, v7}, Landroid/content/pm/PackageParser;->validateName(Ljava/lang/String;Z)Ljava/lang/String;
  
+     move-result-object v4
+ 
+     invoke-static {v1, v4}, Landroid/content/pm/PackageParser$Injector;->filterNameError(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+ 
      move-result-object v0
  
      .local v0, nameError:Ljava/lang/String;
***************
*** 10907,10913 ****
  
      .restart local v6       #p:Landroid/content/pm/PackageParser$Provider;
      :cond_2
!     const/4 v15, 0x0
  
      .local v15, providerExportedDefault:Z
      move-object/from16 v0, p1
--- 10928,10934 ----
  
      .restart local v6       #p:Landroid/content/pm/PackageParser$Provider;
      :cond_2
!     const/4 v15, 0x1
  
      .local v15, providerExportedDefault:Z
      move-object/from16 v0, p1
***************
*** 13336,13341 ****
--- 13357,13376 ----
      .local v13, localCerts:[Ljava/security/cert/Certificate;
      if-nez v13, :cond_5
  
+     move-object/from16 v0, p1
+ 
+     iget-object v0, v0, Landroid/content/pm/PackageParser$Package;->packageName:Ljava/lang/String;
+ 
+     move-object/from16 v18, v0
+ 
+     move-object/from16 v0, v18
+ 
+     invoke-static {v12, v0}, Lmiui/content/pm/ExtraPackageManager;->isTrustedAppEntry(Ljava/util/jar/JarEntry;Ljava/lang/String;)Z
+ 
+     move-result v18
+ 
+     if-nez v18, :cond_3
+ 
      const-string v18, "PackageParser"
  
      new-instance v19, Ljava/lang/StringBuilder;
***************
*** 14021,14027 ****
      .local v25, cookie:I
      if-eqz v25, :cond_4
  
!     new-instance v32, Landroid/content/res/Resources;
  
      const/4 v6, 0x0
  
--- 14056,14062 ----
      .local v25, cookie:I
      if-eqz v25, :cond_4
  
!     new-instance v32, Landroid/content/res/MiuiResources;
  
      const/4 v6, 0x0
  
***************
*** 14029,14035 ****
  
      move-object/from16 v1, p3
  
!     invoke-direct {v0, v5, v1, v6}, Landroid/content/res/Resources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;)V
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
  
--- 14064,14070 ----
  
      move-object/from16 v1, p3
  
!     invoke-direct {v0, v5, v1, v6}, Landroid/content/res/MiuiResources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;)V
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
  
