*** ./smali/android/content/pm/ResolveInfo.smali	2013-11-18 12:21:47.190342859 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/content/pm/ResolveInfo.smali	2013-11-18 12:18:50.882339549 +0800
***************
*** 9,15 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/content/pm/ResolveInfo$DisplayNameComparator;
      }
  .end annotation
  
--- 9,16 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/content/pm/ResolveInfo$DisplayNameComparator;,
!         Landroid/content/pm/ResolveInfo$Injector;
      }
  .end annotation
  
***************
*** 717,722 ****
--- 718,727 ----
      move-result-object v2
  
      .local v2, dr:Landroid/graphics/drawable/Drawable;
+     invoke-static {v2}, Lmiui/content/res/IconCustomizer;->generateIconStyleDrawable(Landroid/graphics/drawable/Drawable;)Landroid/graphics/drawable/BitmapDrawable;
+ 
+     move-result-object v2
+ 
      if-eqz v2, :cond_1
  
      .end local v2           #dr:Landroid/graphics/drawable/Drawable;
***************
*** 742,750 ****
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
      iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
  
!     invoke-virtual {p1, v3, v4, v0}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
--- 747,757 ----
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
+     iget-object v5, v1, Landroid/content/pm/ComponentInfo;->name:Ljava/lang/String;
+ 
      iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
  
!     invoke-static {p1, v3, v5, v4, v0}, Landroid/app/MiuiThemeHelper;->getDrawable(Landroid/content/pm/PackageManager;Ljava/lang/String;Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
