*** ./smali/com/android/internal/policy/impl/KeyguardStatusViewManager$2.smali	2013-11-08 15:41:55.896921503 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/KeyguardStatusViewManager$2.smali	2013-11-08 15:39:03.452921524 +0800
***************
*** 15,21 ****
  
  
  # static fields
! .field static final synthetic $SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
  .field static final synthetic $SwitchMap$com$android$internal$telephony$IccCardConstants$State:[I
  
--- 15,21 ----
  
  
  # static fields
! .field static final synthetic $SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
  .field static final synthetic $SwitchMap$com$android$internal$telephony$IccCardConstants$State:[I
  
***************
*** 33,42 ****
  
      new-array v0, v0, [I
  
!     sput-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      :try_start_0
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->Normal:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
--- 33,42 ----
  
      new-array v0, v0, [I
  
!     sput-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      :try_start_0
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->Normal:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
***************
*** 52,58 ****
  
      :goto_0
      :try_start_1
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->NetworkLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
--- 52,58 ----
  
      :goto_0
      :try_start_1
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->NetworkLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
***************
*** 68,74 ****
  
      :goto_1
      :try_start_2
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimMissing:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
--- 68,74 ----
  
      :goto_1
      :try_start_2
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimMissing:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
***************
*** 84,90 ****
  
      :goto_2
      :try_start_3
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimPermDisabled:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
--- 84,90 ----
  
      :goto_2
      :try_start_3
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimPermDisabled:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
***************
*** 100,106 ****
  
      :goto_3
      :try_start_4
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimMissingLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
--- 100,106 ----
  
      :goto_3
      :try_start_4
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimMissingLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
***************
*** 116,122 ****
  
      :goto_4
      :try_start_5
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
--- 116,122 ----
  
      :goto_4
      :try_start_5
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
***************
*** 132,138 ****
  
      :goto_5
      :try_start_6
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$keyguard_obsolete$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimPukLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
--- 132,138 ----
  
      :goto_5
      :try_start_6
!     sget-object v0, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$2;->$SwitchMap$com$android$internal$policy$impl$KeyguardStatusViewManager$StatusMode:[I
  
      sget-object v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->SimPukLocked:Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
