*** ./smali/com/android/internal/policy/impl/KeyguardViewManager.smali	2013-11-18 12:22:21.442343502 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/KeyguardViewManager.smali	2013-11-18 12:19:24.298340176 +0800
***************
*** 10,16 ****
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/policy/impl/KeyguardViewManager$KeyguardViewHost;,
!         Lcom/android/internal/policy/impl/KeyguardViewManager$ShowListener;
      }
  .end annotation
  
--- 10,17 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/policy/impl/KeyguardViewManager$KeyguardViewHost;,
!         Lcom/android/internal/policy/impl/KeyguardViewManager$ShowListener;,
!         Lcom/android/internal/policy/impl/KeyguardViewManager$Injector;
      }
  .end annotation
  
***************
*** 56,62 ****
  .end method
  
  .method public constructor <init>(Landroid/content/Context;Landroid/view/ViewManager;Lcom/android/internal/policy/impl/KeyguardViewCallback;Lcom/android/internal/policy/impl/KeyguardViewProperties;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;)V
!     .locals 1
      .parameter "context"
      .parameter "viewManager"
      .parameter "callback"
--- 57,63 ----
  .end method
  
  .method public constructor <init>(Landroid/content/Context;Landroid/view/ViewManager;Lcom/android/internal/policy/impl/KeyguardViewCallback;Lcom/android/internal/policy/impl/KeyguardViewProperties;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;)V
!     .locals 2
      .parameter "context"
      .parameter "viewManager"
      .parameter "callback"
***************
*** 72,78 ****
  
      iput-boolean v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mScreenOn:Z
  
!     iput-object p1, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mContext:Landroid/content/Context;
  
      iput-object p2, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mViewManager:Landroid/view/ViewManager;
  
--- 73,85 ----
  
      iput-boolean v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mScreenOn:Z
  
!     new-instance v0, Landroid/view/ContextThemeWrapper;
! 
!     const v1, 0x103006b
! 
!     invoke-direct {v0, p1, v1}, Landroid/view/ContextThemeWrapper;-><init>(Landroid/content/Context;I)V
! 
!     iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mContext:Landroid/content/Context;
  
      iput-object p2, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mViewManager:Landroid/view/ViewManager;
  
***************
*** 97,102 ****
--- 104,180 ----
  
  
  # virtual methods
+ .method public declared-synchronized dismiss()V
+     .locals 1
+ 
+     .prologue
+     monitor-enter p0
+ 
+     :try_start_0
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mScreenOn:Z
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     if-eqz v0, :cond_0
+ 
+     :cond_0
+     monitor-exit p0
+ 
+     return-void
+ 
+     :catchall_0
+     move-exception v0
+ 
+     monitor-exit p0
+ 
+     throw v0
+ .end method
+ 
+ .method getKeyguardView()Lcom/android/internal/policy/impl/KeyguardViewBase;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mKeyguardView:Lcom/android/internal/policy/impl/KeyguardViewBase;
+ 
+     return-object v0
+ .end method
+ 
+ .method getKeyguardViewProperties()Lcom/android/internal/policy/impl/KeyguardViewProperties;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mKeyguardViewProperties:Lcom/android/internal/policy/impl/KeyguardViewProperties;
+ 
+     return-object v0
+ .end method
+ 
+ .method getWindowLayoutParams()Landroid/view/WindowManager$LayoutParams;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mWindowLayoutParams:Landroid/view/WindowManager$LayoutParams;
+ 
+     return-object v0
+ .end method
+ 
+ .method getViewManager()Landroid/view/ViewManager;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mViewManager:Landroid/view/ViewManager;
+ 
+     return-object v0
+ .end method
+ 
+ .method getKeyguardHost()Landroid/widget/FrameLayout;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mKeyguardHost:Landroid/widget/FrameLayout;
+ 
+     return-object v0
+ .end method
+ 
  .method public declared-synchronized hide()V
      .locals 5
  
***************
*** 131,137 ****
  
      invoke-direct {v2, p0, v0}, Lcom/android/internal/policy/impl/KeyguardViewManager$2;-><init>(Lcom/android/internal/policy/impl/KeyguardViewManager;Lcom/android/internal/policy/impl/KeyguardViewBase;)V
  
!     const-wide/16 v3, 0x1f4
  
      invoke-virtual {v1, v2, v3, v4}, Landroid/widget/FrameLayout;->postDelayed(Ljava/lang/Runnable;J)Z
      :try_end_0
--- 209,215 ----
  
      invoke-direct {v2, p0, v0}, Lcom/android/internal/policy/impl/KeyguardViewManager$2;-><init>(Lcom/android/internal/policy/impl/KeyguardViewManager;Lcom/android/internal/policy/impl/KeyguardViewBase;)V
  
!     const-wide/16 v3, 0x0
  
      invoke-virtual {v1, v2, v3, v4}, Landroid/widget/FrameLayout;->postDelayed(Ljava/lang/Runnable;J)Z
      :try_end_0
***************
*** 503,508 ****
--- 581,592 ----
  
      iput v1, v0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
+     iget v1, v0, Landroid/view/WindowManager$LayoutParams;->inputFeatures:I
+ 
+     or-int/lit8 v1, v1, 0x4
+ 
+     iput v1, v0, Landroid/view/WindowManager$LayoutParams;->inputFeatures:I
+ 
      const-string v1, "Keyguard"
  
      invoke-virtual {v0, v1}, Landroid/view/WindowManager$LayoutParams;->setTitle(Ljava/lang/CharSequence;)V
***************
*** 585,630 ****
  
      .end local v0           #lp:Landroid/view/ViewGroup$LayoutParams;
      :cond_6
!     const/high16 v9, 0x60
! 
!     .local v9, visFlags:I
!     sget-object v1, Lcom/android/internal/policy/impl/KeyguardViewManager;->TAG:Ljava/lang/String;
! 
!     new-instance v2, Ljava/lang/StringBuilder;
! 
!     invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
! 
!     const-string v3, "KGVM: Set visibility on "
! 
!     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
! 
!     move-result-object v2
! 
!     iget-object v3, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mKeyguardHost:Landroid/widget/FrameLayout;
! 
!     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
! 
!     move-result-object v2
! 
!     const-string v3, " to "
! 
!     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
! 
!     move-result-object v2
! 
!     invoke-virtual {v2, v9}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
! 
!     move-result-object v2
! 
!     invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
! 
!     move-result-object v2
! 
!     invoke-static {v1, v2}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mKeyguardHost:Landroid/widget/FrameLayout;
  
!     invoke-virtual {v1, v9}, Landroid/widget/FrameLayout;->setSystemUiVisibility(I)V
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mViewManager:Landroid/view/ViewManager;
  
--- 669,679 ----
  
      .end local v0           #lp:Landroid/view/ViewGroup$LayoutParams;
      :cond_6
!     iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mWindowLayoutParams:Landroid/view/WindowManager$LayoutParams;
  
!     const-wide/16 v2, 0x2710
  
!     iput-wide v2, v1, Landroid/view/WindowManager$LayoutParams;->userActivityTimeout:J
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewManager;->mViewManager:Landroid/view/ViewManager;
  
