*** ./smali/com/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer.smali	2013-11-18 12:22:21.862343509 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer.smali	2013-11-18 12:19:24.666340183 +0800
***************
*** 32,38 ****
  
  .field private final mAccountManager:Landroid/accounts/AccountManager;
  
! .field private final mAccounts:[Landroid/accounts/Account;
  
  .field final synthetic this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
--- 32,38 ----
  
  .field private final mAccountManager:Landroid/accounts/AccountManager;
  
! .field private mAccounts:[Landroid/accounts/Account;
  
  .field final synthetic this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
***************
*** 58,63 ****
--- 58,83 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;->mAccounts:[Landroid/accounts/Account;
  
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;->mAccounts:[Landroid/accounts/Account;
+ 
+     if-eqz v0, :cond_miui_0
+ 
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;->mAccounts:[Landroid/accounts/Account;
+ 
+     array-length v0, v0
+ 
+     if-nez v0, :cond_miui_1
+ 
+     :cond_miui_0
+     const-string v0, "com.xiaomi"
+ 
+     invoke-virtual {p2, v0}, Landroid/accounts/AccountManager;->getAccountsByType(Ljava/lang/String;)[Landroid/accounts/Account;
+ 
+     move-result-object v0
+ 
+     iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;->mAccounts:[Landroid/accounts/Account;
+ 
+     :cond_miui_1
      return-void
  .end method
  
***************
*** 118,124 ****
  
      move-result-object v0
  
!     instance-of v0, v0, Lcom/android/internal/policy/impl/PatternUnlockScreen;
  
      if-eqz v0, :cond_1
  
--- 138,144 ----
  
      move-result-object v0
  
!     instance-of v0, v0, Lcom/android/internal/policy/impl/MiuiCommonUnlockScreen;
  
      if-eqz v0, :cond_1
  
***************
*** 129,135 ****
  
      move-result-object v0
  
!     check-cast v0, Lcom/android/internal/policy/impl/PatternUnlockScreen;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
--- 149,155 ----
  
      move-result-object v0
  
!     check-cast v0, Lcom/android/internal/policy/impl/MiuiCommonUnlockScreen;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
***************
*** 138,144 ****
  
      move-result v1
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/PatternUnlockScreen;->setEnableFallback(Z)V
  
      goto :goto_0
  
--- 158,164 ----
  
      move-result v1
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/MiuiCommonUnlockScreen;->setEnableFallback(Z)V
  
      goto :goto_0
  
