*** ./smali/com/android/internal/policy/impl/PhoneFallbackEventHandler.smali	2013-11-10 11:52:28.217671375 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/PhoneFallbackEventHandler.smali	2013-11-10 11:53:47.313673400 +0800
***************
*** 253,258 ****
--- 253,265 ----
      return-object v0
  .end method
  
+ .method handleCameraKeyEvent()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
  .method onKeyDown(ILandroid/view/KeyEvent;)Z
      .locals 13
      .parameter "keyCode"
***************
*** 677,682 ****
--- 684,691 ----
  
      if-nez v3, :cond_2
  
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->handleCameraKeyEvent()V
+ 
      goto :goto_0
  
      :sswitch_3
