*** ./smali/com/android/internal/policy/impl/PhoneWindow.smali	2013-11-08 15:41:53.056921502 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/PhoneWindow.smali	2013-11-08 15:39:00.388921524 +0800
***************
*** 16,22 ****
          Lcom/android/internal/policy/impl/PhoneWindow$DecorView;,
          Lcom/android/internal/policy/impl/PhoneWindow$ActionMenuPresenterCallback;,
          Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;,
!         Lcom/android/internal/policy/impl/PhoneWindow$WindowManagerHolder;
      }
  .end annotation
  
--- 16,23 ----
          Lcom/android/internal/policy/impl/PhoneWindow$DecorView;,
          Lcom/android/internal/policy/impl/PhoneWindow$ActionMenuPresenterCallback;,
          Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;,
!         Lcom/android/internal/policy/impl/PhoneWindow$WindowManagerHolder;,
!         Lcom/android/internal/policy/impl/PhoneWindow$Injector;
      }
  .end annotation
  
***************
*** 1194,1199 ****
--- 1195,1204 ----
      invoke-virtual {v6}, Lcom/android/internal/widget/ActionBarView;->initIndeterminateProgress()V
  
      :cond_7
+     iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindow;->mActionBar:Lcom/android/internal/widget/ActionBarView;
+ 
+     invoke-static {p0, v6}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->handleStartingWindow(Lcom/android/internal/policy/impl/PhoneWindow;Lcom/android/internal/widget/ActionBarView;)V
+ 
      const/4 v2, 0x0
  
      .local v2, splitActionBar:Z
***************
*** 1364,1370 ****
  
      const/4 v3, 0x0
  
!     invoke-static {v2, v3}, Landroid/graphics/drawable/Drawable;->createFromStream(Ljava/io/InputStream;Ljava/lang/String;)Landroid/graphics/drawable/Drawable;
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
--- 1369,1375 ----
  
      const/4 v3, 0x0
  
!     invoke-static {v3, v3, v2, v3, v3}, Landroid/graphics/drawable/Drawable;->createFromResourceStreamAndCloseStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
***************
*** 1766,1771 ****
--- 1771,1782 ----
  
      iput v4, v2, Landroid/view/WindowManager$LayoutParams;->windowAnimations:I
  
+     move-object/from16 v0, p0
+ 
+     move-object/from16 v1, v17
+ 
+     invoke-static {v0, v1, v2}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->handleAppLayoutParams(Lcom/android/internal/policy/impl/PhoneWindow;Landroid/view/WindowManager;Landroid/view/WindowManager$LayoutParams;)V
+ 
      move-object/from16 v0, p1
  
      iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->decorView:Lcom/android/internal/policy/impl/PhoneWindow$DecorView;
***************
*** 1829,1835 ****
  
      iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->createdPanelView:Landroid/view/View;
  
!     if-eqz v4, :cond_a
  
      move-object/from16 v0, p1
  
--- 1840,1846 ----
  
      iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->createdPanelView:Landroid/view/View;
  
!     if-eqz v4, :cond_miui_0
  
      move-object/from16 v0, p1
  
***************
*** 1853,1858 ****
--- 1864,1896 ----
      goto/16 :goto_5
  
      .local v2, lp:Landroid/view/WindowManager$LayoutParams;
+     :cond_miui_0
+     move-object/from16 v0, p1
+ 
+     iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->decorView:Lcom/android/internal/policy/impl/PhoneWindow$DecorView;
+ 
+     invoke-virtual {v4}, Lcom/android/internal/policy/impl/PhoneWindow$DecorView;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
+ 
+     move-result-object v2
+ 
+     .restart local v2       #lp:Landroid/view/ViewGroup$LayoutParams;
+     if-eqz v2, :cond_miui_1
+ 
+     iget v3, v2, Landroid/view/ViewGroup$LayoutParams;->width:I
+ 
+     :cond_miui_1
+     move-object/from16 v0, p1
+ 
+     iget-object v4, v0, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->decorView:Lcom/android/internal/policy/impl/PhoneWindow$DecorView;
+ 
+     invoke-virtual {v4}, Lcom/android/internal/policy/impl/PhoneWindow$DecorView;->getChildCount()I
+ 
+     move-result v4
+ 
+     if-nez v4, :cond_a
+ 
+     goto/16 :goto_0
+     
      :cond_f
      move-object/from16 v0, p1
  
***************
*** 3121,3127 ****
  
      if-eqz v19, :cond_19
  
!     const/16 v19, -0x2
  
      const/16 v20, -0x2
  
--- 3159,3171 ----
  
      if-eqz v19, :cond_19
  
!     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/PhoneWindow;->getContext()Landroid/content/Context;
! 
!     move-result-object v19
! 
!     invoke-static/range {v19 .. v19}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->getFloatingWindowWidth(Landroid/content/Context;)I
! 
!     move-result v19
  
      const/16 v20, -0x2
  
***************
*** 4096,4102 ****
  
      move/from16 v1, v19
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/PhoneWindow;->clearFlags(I)V
  
      goto/16 :goto_6
  
--- 4140,4146 ----
  
      move/from16 v1, v19
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/PhoneWindow;->addFlags(I)V
  
      goto/16 :goto_6
  
***************
*** 4247,4260 ****
  
      if-eqz v19, :cond_26
  
!     const v10, 0x10900b7
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
  
      .end local v10           #layoutResource:I
      :cond_26
!     const v10, 0x10900b6
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
--- 4291,4316 ----
  
      if-eqz v19, :cond_26
  
!     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/PhoneWindow;->getContext()Landroid/content/Context;
! 
!     move-result-object v19
! 
!     invoke-static/range {v19 .. v19}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->getActionBarOverlayResourceId(Landroid/content/Context;)I
! 
!     move-result v10
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
  
      .end local v10           #layoutResource:I
      :cond_26
!     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/PhoneWindow;->getContext()Landroid/content/Context;
! 
!     move-result-object v19
! 
!     invoke-static/range {v19 .. v19}, Lcom/android/internal/policy/impl/PhoneWindow$Injector;->getActionBarResourceId(Landroid/content/Context;)I
! 
!     move-result v10
  
      .restart local v10       #layoutResource:I
      goto/16 :goto_8
***************
*** 4468,4473 ****
--- 4524,4538 ----
      return-object v4
  .end method
  
+ .method getActionBarView()Lcom/android/internal/widget/ActionBarView;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindow;->mActionBar:Lcom/android/internal/widget/ActionBarView;
+ 
+     return-object v0
+ .end method
+ 
  .method getAudioManager()Landroid/media/AudioManager;
      .locals 2
  
