*** ./smali/com/android/internal/policy/impl/PhoneWindowManager$SettingsObserver.smali	2013-11-10 11:52:28.805671390 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/PhoneWindowManager$SettingsObserver.smali	2013-11-10 11:53:48.009673418 +0800
***************
*** 115,120 ****
--- 115,128 ----
  
      invoke-virtual {v0, v1, v3, p0, v2}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;I)V
  
+     const-string v1, "screensaver_enabled"
+ 
+     invoke-static {v1}, Landroid/provider/Settings$Secure;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v0, v1, v3, p0, v2}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;I)V
+ 
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-virtual {v1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->updateSettings()V
