*** ./smali/android/util/TypedValue.smali	2013-11-18 12:21:52.766342963 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/util/TypedValue.smali	2013-11-18 12:18:56.110339647 +0800
***************
*** 88,93 ****
--- 88,95 ----
  
  .field public static final TYPE_STRING:I = 0x3
  
+ .field public static miui_font_scale:F
+ 
  
  # instance fields
  .field public assetCookie:I
***************
*** 118,123 ****
--- 120,129 ----
  
      const/4 v3, 0x0
  
+     const/high16 v0, 0x3f80
+ 
+     sput v0, Landroid/util/TypedValue;->miui_font_scale:F
+ 
      new-array v0, v6, [F
  
      fill-array-data v0, :array_0
***************
*** 221,229 ****
      goto :goto_0
  
      :pswitch_2
!     iget v0, p2, Landroid/util/DisplayMetrics;->scaledDensity:F
  
!     mul-float/2addr p1, v0
  
      goto :goto_0
  
--- 227,235 ----
      goto :goto_0
  
      :pswitch_2
!     invoke-static {p1, p2}, Landroid/util/TypedValue;->miuiScale(FLandroid/util/DisplayMetrics;)F
  
!     move-result p1
  
      goto :goto_0
  
***************
*** 779,784 ****
--- 785,821 ----
      .end packed-switch
  .end method
  
+ .method private static miuiScale(FLandroid/util/DisplayMetrics;)F
+     .locals 2
+     .parameter "value"
+     .parameter "metrics"
+ 
+     .prologue
+     const/high16 v0, 0x41a0
+ 
+     cmpl-float v0, p0, v0
+ 
+     if-ltz v0, :cond_0
+ 
+     iget v0, p1, Landroid/util/DisplayMetrics;->density:F
+ 
+     mul-float/2addr v0, p0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     iget v0, p1, Landroid/util/DisplayMetrics;->scaledDensity:F
+ 
+     mul-float/2addr v0, p0
+ 
+     sget v1, Landroid/util/TypedValue;->miui_font_scale:F
+ 
+     mul-float/2addr v0, v1
+ 
+     goto :goto_0
+ .end method
+ 
  
  # virtual methods
  .method public final coerceToString()Ljava/lang/CharSequence;
