*** ./smali/com/android/internal/content/PackageMonitor.smali	2013-11-18 12:22:13.138343346 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/com/android/internal/content/PackageMonitor.smali	2013-11-18 12:19:16.026340021 +0800
***************
*** 1291,1296 ****
--- 1291,1309 ----
      return-void
  .end method
  
+ .method public register(Landroid/content/Context;Z)V
+     .locals 1
+     .parameter "context"
+     .parameter "externalStorage"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     invoke-virtual {p0, p1, v0, p2}, Lcom/android/internal/content/PackageMonitor;->register(Landroid/content/Context;Landroid/os/Looper;Z)V
+ 
+     return-void
+ .end method
+ 
  .method public unregister()V
      .locals 2
  
