*** ./smali/android/preference/PreferenceFrameLayout$LayoutParams.smali	2013-11-18 12:22:01.950343136 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/preference/PreferenceFrameLayout$LayoutParams.smali	2013-11-18 12:19:05.218339818 +0800
***************
*** 59,64 ****
--- 59,72 ----
  
      iput-boolean v1, p0, Landroid/preference/PreferenceFrameLayout$LayoutParams;->removeBorders:Z
  
+     iget-boolean v1, p0, Landroid/preference/PreferenceFrameLayout$LayoutParams;->removeBorders:Z
+ 
+     invoke-static {p1, v1}, Landroid/preference/PreferenceFrameLayout$Injector;->isRemoveBorders(Landroid/content/Context;Z)Z
+ 
+     move-result v1
+ 
+     iput-boolean v1, p0, Landroid/preference/PreferenceFrameLayout$LayoutParams;->removeBorders:Z
+ 
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
      return-void
