*** ./smali/android/preference/Preference.smali	2013-11-18 12:22:02.338343143 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/preference/Preference.smali	2013-11-18 12:19:05.594339825 +0800
***************
*** 11,16 ****
--- 11,17 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/preference/Preference$BaseSavedState;,
+         Landroid/preference/Preference$Injector;,
          Landroid/preference/Preference$OnPreferenceChangeInternalListener;,
          Landroid/preference/Preference$OnPreferenceClickListener;,
          Landroid/preference/Preference$OnPreferenceChangeListener;
***************
*** 87,92 ****
--- 88,95 ----
  
  .field private mPreferenceManager:Landroid/preference/PreferenceManager;
  
+ .field mPreferenceParent:Landroid/preference/PreferenceGroup;
+ 
  .field private mRequiresKey:Z
  
  .field private mSelectable:Z
***************
*** 1562,1567 ****
--- 1565,1572 ----
  
      const/4 v5, 0x0
  
+     invoke-static {p0, p1}, Landroid/preference/Preference$Injector;->onBindView(Landroid/preference/Preference;Landroid/view/View;)V
+ 
      const v7, 0x1020016
  
      invoke-virtual {p1, v7}, Landroid/view/View;->findViewById(I)Landroid/view/View;
***************
*** 2529,2534 ****
--- 2534,2549 ----
      return-void
  .end method
  
+ .method public setParent(Landroid/preference/PreferenceGroup;)V
+     .locals 0
+     .parameter "parent"
+ 
+     .prologue
+     iput-object p1, p0, Landroid/preference/Preference;->mPreferenceParent:Landroid/preference/PreferenceGroup;
+ 
+     return-void
+ .end method
+ 
  .method public setPersistent(Z)V
      .locals 0
      .parameter "persistent"
