*** ./smali/com/android/internal/widget/ActionBarContainer.smali	2013-11-08 15:41:46.804921504 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/ActionBarContainer.smali	2013-11-08 15:38:54.236921525 +0800
***************
*** 128,133 ****
--- 128,167 ----
      goto :goto_0
  .end method
  
+ .method private shouldPutTabsOnTop()Z
+     .locals 2
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     iget-object v1, p0, Lcom/android/internal/widget/ActionBarContainer;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v1}, Lmiui/util/UiUtils;->isV5Ui(Landroid/content/Context;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_1
+ 
+     :cond_0
+     :goto_0
+     return v0
+ 
+     :cond_1
+     iget-object v1, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
+ 
+     invoke-virtual {v1}, Lcom/android/internal/widget/ActionBarView;->getDisplayOptions()I
+ 
+     move-result v1
+ 
+     and-int/lit8 v1, v1, 0x2
+ 
+     if-nez v1, :cond_0
+ 
+     const/4 v0, 0x1
+ 
+     goto :goto_0
+ .end method
+ 
  
  # virtual methods
  .method protected drawableStateChanged()V
***************
*** 202,207 ****
--- 236,250 ----
      return-void
  .end method
  
+ .method protected getActionBarView()Lcom/android/internal/widget/ActionBarView;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
+ 
+     return-object v0
+ .end method
+ 
  .method public getTabContainer()Landroid/view/View;
      .locals 1
  
***************
*** 211,216 ****
--- 254,268 ----
      return-object v0
  .end method
  
+ .method protected isSplit()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/widget/ActionBarContainer;->mIsSplit:Z
+ 
+     return v0
+ .end method
+ 
  .method public jumpDrawablesToCurrentState()V
      .locals 1
  
***************
*** 421,435 ****
      move-result v7
  
      .local v7, tabHeight:I
!     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
! 
!     invoke-virtual {v8}, Lcom/android/internal/widget/ActionBarView;->getDisplayOptions()I
  
      move-result v8
  
!     and-int/lit8 v8, v8, 0x2
! 
!     if-nez v8, :cond_7
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarContainer;->getChildCount()I
  
--- 473,483 ----
      move-result v7
  
      .local v7, tabHeight:I
!     invoke-direct {p0}, Lcom/android/internal/widget/ActionBarContainer;->shouldPutTabsOnTop()Z
  
      move-result v8
  
!     if-eqz v8, :cond_7
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ActionBarContainer;->getChildCount()I
  
***************
*** 477,482 ****
--- 525,540 ----
      :cond_2
      iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
  
+     invoke-virtual {v8}, Lcom/android/internal/widget/ActionBarView;->getVisibility()I
+ 
+     move-result v8
+ 
+     const/16 v9, 0x8
+ 
+     if-eq v8, v9, :cond_0
+ 
+     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
+ 
      invoke-virtual {v8}, Lcom/android/internal/widget/ActionBarView;->isCollapsed()Z
  
      move-result v8
