*** ./smali/com/android/internal/widget/ScrollingTabContainerView.smali	2013-11-18 12:22:14.734343376 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/ScrollingTabContainerView.smali	2013-11-18 12:19:17.598340050 +0800
***************
*** 12,18 ****
          Lcom/android/internal/widget/ScrollingTabContainerView$VisibilityAnimListener;,
          Lcom/android/internal/widget/ScrollingTabContainerView$TabClickListener;,
          Lcom/android/internal/widget/ScrollingTabContainerView$TabAdapter;,
!         Lcom/android/internal/widget/ScrollingTabContainerView$TabView;
      }
  .end annotation
  
--- 12,19 ----
          Lcom/android/internal/widget/ScrollingTabContainerView$VisibilityAnimListener;,
          Lcom/android/internal/widget/ScrollingTabContainerView$TabClickListener;,
          Lcom/android/internal/widget/ScrollingTabContainerView$TabAdapter;,
!         Lcom/android/internal/widget/ScrollingTabContainerView$TabView;,
!         Lcom/android/internal/widget/ScrollingTabContainerView$Injector;
      }
  .end annotation
  
***************
*** 667,672 ****
--- 668,691 ----
      goto :goto_0
  .end method
  
+ .method protected getSelectedTabIndex()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/widget/ScrollingTabContainerView;->mSelectedTabIndex:I
+ 
+     return v0
+ .end method
+ 
+ .method protected getTabLayout()Landroid/widget/LinearLayout;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/widget/ScrollingTabContainerView;->mTabLayout:Landroid/widget/LinearLayout;
+ 
+     return-object v0
+ .end method
+ 
  .method public onAttachedToWindow()V
      .locals 1
  
