*** ./smali/com/android/internal/statusbar/IStatusBar$Stub.smali	2013-11-18 12:22:13.874343359 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/com/android/internal/statusbar/IStatusBar$Stub.smali	2013-11-18 12:19:16.750340034 +0800
***************
*** 50,55 ****
--- 50,57 ----
  
  .field static final TRANSACTION_setImeWindowStatus:I = 0xc
  
+ .field static final TRANSACTION_setStatus:I = 0x11
+ 
  .field static final TRANSACTION_setSystemUiVisibility:I = 0xa
  
  .field static final TRANSACTION_toggleRecentApps:I = 0xe
***************
*** 486,491 ****
--- 488,535 ----
  
      goto/16 :goto_0
  
+     :sswitch_11
+     const-string v4, "com.android.internal.statusbar.IStatusBar"
+ 
+     invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v0
+ 
+     .local v0, _arg0:I
+     invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     .local v1, _arg1:Ljava/lang/String;
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v4
+ 
+     if-eqz v4, :cond_6
+ 
+     sget-object v4, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
+ 
+     invoke-interface {v4, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
+ 
+     move-result-object v2
+ 
+     check-cast v2, Landroid/os/Bundle;
+ 
+     .local v2, _arg2:Landroid/os/Bundle;
+     :goto_7
+     invoke-virtual {p0, v0, v1, v2}, Lcom/android/internal/statusbar/IStatusBar$Stub;->setStatus(ILjava/lang/String;Landroid/os/Bundle;)V
+ 
+     goto/16 :goto_0
+ 
+     .end local v2           #_arg2:Landroid/os/Bundle;
+     :cond_6
+     const/4 v2, 0x0
+ 
+     .restart local v2       #_arg2:Landroid/os/Bundle;
+     goto :goto_7
+ 
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
***************
*** 504,509 ****
--- 548,554 ----
          0xe -> :sswitch_e
          0xf -> :sswitch_f
          0x10 -> :sswitch_10
+         0x11 -> :sswitch_11
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
