*** ./smali/com/android/internal/statusbar/IStatusBar.smali	2013-11-10 11:53:29.277672939 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/com/android/internal/statusbar/IStatusBar.smali	2013-11-10 11:54:49.033674981 +0800
***************
*** 111,116 ****
--- 111,124 ----
      .end annotation
  .end method
  
+ .method public abstract setStatus(ILjava/lang/String;Landroid/os/Bundle;)V
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ .end method
+ 
  .method public abstract setSystemUiVisibility(II)V
      .annotation system Ldalvik/annotation/Throws;
          value = {
