*** ./smali/com/android/internal/telephony/gsm/SpnOverride.smali	2013-11-08 15:41:49.872921504 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/gsm/SpnOverride.smali	2013-11-08 15:38:57.184921523 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "SpnOverride.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Lcom/android/internal/telephony/gsm/SpnOverride$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field static final LOG_TAG:Ljava/lang/String; = "GSM"
  
***************
*** 97,102 ****
--- 105,113 ----
  
      .end local v2           #name:Ljava/lang/String;
      .end local v4           #parser:Lorg/xmlpull/v1/XmlPullParser;
+     :goto_miui_0
+     invoke-static {v6}, Lcom/android/internal/telephony/gsm/SpnOverride$Injector;->closeSpnReader(Ljava/io/FileReader;)V
+ 
      .end local v6           #spnReader:Ljava/io/FileReader;
      :goto_1
      return-void
***************
*** 208,214 ****
  
      invoke-static {v7, v8}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
!     goto :goto_1
  
      .end local v1           #e:Lorg/xmlpull/v1/XmlPullParserException;
      :catch_2
--- 219,225 ----
  
      invoke-static {v7, v8}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
!     goto :goto_miui_0
  
      .end local v1           #e:Lorg/xmlpull/v1/XmlPullParserException;
      :catch_2
***************
*** 237,243 ****
  
      invoke-static {v7, v8}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
!     goto :goto_1
  .end method
  
  
--- 248,254 ----
  
      invoke-static {v7, v8}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
!     goto :goto_miui_0
  .end method
  
  
***************
*** 260,265 ****
--- 271,292 ----
      .locals 1
      .parameter "carrier"
  
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/gsm/SpnOverride;->CarrierSpnMap:Ljava/util/HashMap;
+ 
+     invoke-virtual {v0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
+ 
+     move-result-object v0
+ 
+     check-cast v0, Ljava/lang/String;
+ 
+     return-object v0
+ .end method
+ 
+ .method public getSpnFromConfig(Ljava/lang/String;)Ljava/lang/String;
+     .locals 1
+     .parameter "carrier"
+ 
      .prologue
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/SpnOverride;->CarrierSpnMap:Ljava/util/HashMap;
  
