*** ./smali/android/view/View.smali	2013-11-18 12:21:48.490342883 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/view/View.smali	2013-11-18 12:18:51.974339569 +0800
***************
*** 37,43 ****
          Landroid/view/View$DragShadowBuilder;,
          Landroid/view/View$OnLayoutChangeListener;,
          Landroid/view/View$ListenerInfo;,
!         Landroid/view/View$TransformationInfo;
      }
  .end annotation
  
--- 37,44 ----
          Landroid/view/View$DragShadowBuilder;,
          Landroid/view/View$OnLayoutChangeListener;,
          Landroid/view/View$ListenerInfo;,
!         Landroid/view/View$TransformationInfo;,
!         Landroid/view/View$Injector;
      }
  .end annotation
  
***************
*** 670,675 ****
--- 671,678 ----
  
  .field mAccessibilityViewId:I
  
+ .field mAdditionalState:I
+ 
  .field private mAnimator:Landroid/view/ViewPropertyAnimator;
  
  .field mAttachInfo:Landroid/view/View$AttachInfo;
***************
*** 709,714 ****
--- 712,719 ----
  
  .field private mFloatingTreeObserver:Landroid/view/ViewTreeObserver;
  
+ .field mHapticEnabledExplicitly:Z
+ 
  .field private mHardwareLayer:Landroid/view/HardwareLayer;
  
  .field private mHasPerformedLongPress:Z
***************
*** 1736,1741 ****
--- 1741,1748 ----
      :goto_0
      iput-object v0, p0, Landroid/view/View;->mInputEventConsistencyVerifier:Landroid/view/InputEventConsistencyVerifier;
  
+     iput v2, p0, Landroid/view/View;->mAdditionalState:I
+ 
      iput-object v1, p0, Landroid/view/View;->mResources:Landroid/content/res/Resources;
  
      return-void
***************
*** 1822,1827 ****
--- 1829,1836 ----
      :goto_0
      iput-object v0, p0, Landroid/view/View;->mInputEventConsistencyVerifier:Landroid/view/InputEventConsistencyVerifier;
  
+     iput v2, p0, Landroid/view/View;->mAdditionalState:I
+ 
      iput-object p1, p0, Landroid/view/View;->mContext:Landroid/content/Context;
  
      if-eqz p1, :cond_0
***************
*** 2676,2681 ****
--- 2685,2704 ----
      goto/16 :goto_1
  
      :pswitch_24
+     const/16 v47, 0x0
+ 
+     move/from16 v0, v47
+ 
+     invoke-virtual {v6, v7, v0}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
+ 
+     move-result v47
+ 
+     move/from16 v0, v47
+ 
+     move-object/from16 v1, p0
+ 
+     iput-boolean v0, v1, Landroid/view/View;->mHapticEnabledExplicitly:Z
+ 
      const/16 v47, 0x1
  
      move/from16 v0, v47
***************
*** 3405,3410 ****
--- 3428,3441 ----
      :cond_f
      invoke-virtual/range {p0 .. p0}, Landroid/view/View;->computeOpaqueFlags()V
  
+     move-object/from16 v0, p0
+ 
+     move-object/from16 v1, p2
+ 
+     move/from16 v2, p3
+ 
+     invoke-static {v0, v1, v2}, Landroid/view/View$Injector;->initializeChildrenSequenceStates(Landroid/view/View;Landroid/util/AttributeSet;I)V
+ 
      return-void
  
      .restart local v9       #bottomPadding:I
***************
*** 13696,13701 ****
--- 13727,13734 ----
  
      invoke-virtual {v0, v1}, Landroid/graphics/drawable/Drawable;->setState([I)Z
  
+     invoke-static {p0, v0}, Landroid/view/View$Injector;->onDrawableStateChanged(Landroid/view/View;Landroid/graphics/drawable/Drawable;)V
+ 
      :cond_0
      return-void
  .end method
***************
*** 13735,13740 ****
--- 13768,13795 ----
      return-void
  .end method
  
+ .method fillAdditionalState([I)[I
+     .locals 2
+     .parameter "states"
+ 
+     .prologue
+     move-object v0, p1
+ 
+     .local v0, newStates:[I
+     iget v1, p0, Landroid/view/View;->mAdditionalState:I
+ 
+     if-eqz v1, :cond_0
+ 
+     iget v1, p0, Landroid/view/View;->mAdditionalState:I
+ 
+     invoke-static {v0, v1}, Lmiui/util/UiUtils;->getViewStates([II)[I
+ 
+     move-result-object v0
+ 
+     :cond_0
+     return-object v0
+ .end method
+ 
  .method public findFocus()Landroid/view/View;
      .locals 1
  
***************
*** 21475,21480 ****
--- 21530,21539 ----
      aget-object v0, v5, v4
  
      .local v0, drawableState:[I
+     invoke-virtual {p0, v0}, Landroid/view/View;->fillAdditionalState([I)[I
+ 
+     move-result-object v0
+ 
      if-eqz p1, :cond_0
  
      if-eqz v0, :cond_c
***************
*** 23474,23479 ****
--- 23533,23540 ----
      iput-object v6, p0, Landroid/view/View;->mPerformClick:Landroid/view/View$PerformClick;
  
      :cond_a
+     invoke-static {p0}, Landroid/view/View$Injector;->performHapticFeedbackOnRelease(Landroid/view/View;)V
+ 
      iget-object v6, p0, Landroid/view/View;->mPerformClick:Landroid/view/View$PerformClick;
  
      invoke-virtual {p0, v6}, Landroid/view/View;->post(Ljava/lang/Runnable;)Z
***************
*** 23592,23597 ****
--- 23653,23660 ----
  
      invoke-direct {p0, v6}, Landroid/view/View;->checkForLongClick(I)V
  
+     invoke-static {p0}, Landroid/view/View$Injector;->performHapticFeedbackOnDown(Landroid/view/View;)V
+ 
      goto/16 :goto_1
  
      .end local v1           #isInScrollingContainer:Z
***************
*** 24523,24528 ****
--- 24586,24597 ----
      return v0
  
      :cond_1
+     invoke-static {p0, p2}, Landroid/view/View$Injector;->isHapticEnabledExplictly(Landroid/view/View;I)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
      and-int/lit8 v1, p2, 0x1
  
      if-nez v1, :cond_2
***************
*** 26681,26686 ****
--- 26750,26776 ----
      goto :goto_1
  .end method
  
+ .method public setAdditionalState(I)V
+     .locals 1
+     .parameter "state"
+ 
+     .prologue
+     iget v0, p0, Landroid/view/View;->mAdditionalState:I
+ 
+     if-eq p1, v0, :cond_0
+ 
+     iput p1, p0, Landroid/view/View;->mAdditionalState:I
+ 
+     const/4 v0, 0x1
+ 
+     invoke-virtual {p0, v0}, Landroid/view/View;->invalidate(Z)V
+ 
+     invoke-virtual {p0}, Landroid/view/View;->refreshDrawableState()V
+ 
+     :cond_0
+     return-void
+ .end method
+ 
  .method public setAlpha(F)V
      .locals 3
      .parameter "alpha"
***************
*** 28722,28727 ****
--- 28812,28819 ----
      :goto_0
      invoke-virtual {p0, v0, v1}, Landroid/view/View;->setFlags(II)V
  
+     iput-boolean p1, p0, Landroid/view/View;->mHapticEnabledExplicitly:Z
+ 
      return-void
  
      :cond_0
