*** ./smali/android/view/ViewGroup.smali	2013-11-18 12:21:49.274342898 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/view/ViewGroup.smali	2013-11-18 12:18:52.574339580 +0800
***************
*** 16,22 ****
          Landroid/view/ViewGroup$TouchTarget;,
          Landroid/view/ViewGroup$MarginLayoutParams;,
          Landroid/view/ViewGroup$LayoutParams;,
!         Landroid/view/ViewGroup$OnHierarchyChangeListener;
      }
  .end annotation
  
--- 16,24 ----
          Landroid/view/ViewGroup$TouchTarget;,
          Landroid/view/ViewGroup$MarginLayoutParams;,
          Landroid/view/ViewGroup$LayoutParams;,
!         Landroid/view/ViewGroup$OnHierarchyChangeListener;,
!         Landroid/view/ViewGroup$Injector;,
!         Landroid/view/ViewGroup$ChildSequenceStateTaggingListener;
      }
  .end annotation
  
***************
*** 116,121 ****
--- 118,125 ----
      .end annotation
  .end field
  
+ .field mChildSequenceStateTaggingListener:Landroid/view/ViewGroup$ChildSequenceStateTaggingListener;
+ 
  .field final mChildTransformation:Landroid/view/animation/Transformation;
  
  .field private mChildren:[Landroid/view/View;
***************
*** 220,225 ****
--- 224,231 ----
  
  .field protected mPersistentDrawingCache:I
  
+ .field mTagChildrenSequenceState:Z
+ 
  .field private mTransition:Landroid/animation/LayoutTransition;
  
  .field private mTransitioningViews:Ljava/util/ArrayList;
***************
*** 547,552 ****
--- 553,560 ----
  
      :cond_1
      :goto_0
+     invoke-static {p0}, Landroid/view/ViewGroup$Injector;->tagChildSequenceState(Landroid/view/ViewGroup;)V
+ 
      return-void
  
      :cond_2
***************
*** 2009,2014 ****
--- 2017,2028 ----
      .end local v2           #animateLayoutChanges:Z
      .end local v3           #attr:I
      :cond_1
+     invoke-static {p1, p2}, Landroid/view/ViewGroup$Injector;->initTagChildSequenceState(Landroid/content/Context;Landroid/util/AttributeSet;)Z
+ 
+     move-result v6
+ 
+     iput-boolean v6, p0, Landroid/view/ViewGroup;->mTagChildrenSequenceState:Z
+ 
      invoke-virtual {v1}, Landroid/content/res/TypedArray;->recycle()V
  
      return-void
***************
*** 2356,2361 ****
--- 2370,2377 ----
  
      :cond_2
      :goto_1
+     invoke-static {p0}, Landroid/view/ViewGroup$Injector;->tagChildSequenceState(Landroid/view/ViewGroup;)V
+ 
      return-void
  
      :cond_3
***************
*** 2496,2501 ****
--- 2512,2519 ----
  
      iput v4, p0, Landroid/view/ViewGroup;->mChildrenCount:I
  
+     invoke-static {p0}, Landroid/view/ViewGroup$Injector;->tagChildSequenceState(Landroid/view/ViewGroup;)V
+ 
      goto :goto_0
  .end method
  
***************
*** 3874,3880 ****
  
      .local v2, i:I
      :goto_0
!     if-ltz v2, :cond_0
  
      aget-object v3, v0, v2
  
--- 3892,3898 ----
  
      .local v2, i:I
      :goto_0
!     if-ltz v2, :cond_miui_0
  
      aget-object v3, v0, v2
  
***************
*** 3885,3899 ****
      add-int/lit8 v2, v2, -0x1
  
      goto :goto_0
  .end method
  
  .method protected detachViewFromParent(I)V
!     .locals 0
      .parameter "index"
  
      .prologue
      invoke-direct {p0, p1}, Landroid/view/ViewGroup;->removeFromArray(I)V
  
      return-void
  .end method
  
--- 3903,3928 ----
      add-int/lit8 v2, v2, -0x1
  
      goto :goto_0
+     
+     :cond_miui_0
+     iget-object v3, p0, Landroid/view/ViewGroup;->mFocused:Landroid/view/View;
+ 
+     invoke-static {p0, v3}, Landroid/view/ViewGroup$Injector;->clearChildFocus(Landroid/view/ViewGroup;Landroid/view/View;)V
+     
+     return-void
  .end method
  
  .method protected detachViewFromParent(I)V
!     .locals 1
      .parameter "index"
  
      .prologue
      invoke-direct {p0, p1}, Landroid/view/ViewGroup;->removeFromArray(I)V
  
+     iget-object v0, p0, Landroid/view/ViewGroup;->mFocused:Landroid/view/View;
+ 
+     invoke-static {p0, v0}, Landroid/view/ViewGroup$Injector;->clearChildFocus(Landroid/view/ViewGroup;Landroid/view/View;)V
+ 
      return-void
  .end method
  
***************
*** 3908,3924 ****
  
      invoke-direct {p0, v0}, Landroid/view/ViewGroup;->removeFromArray(I)V
  
      return-void
  .end method
  
  .method protected detachViewsFromParent(II)V
!     .locals 0
      .parameter "start"
      .parameter "count"
  
      .prologue
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup;->removeFromArray(II)V
  
      return-void
  .end method
  
--- 3937,3961 ----
  
      invoke-direct {p0, v0}, Landroid/view/ViewGroup;->removeFromArray(I)V
  
+     iget-object v0, p0, Landroid/view/ViewGroup;->mFocused:Landroid/view/View;
+ 
+     invoke-static {p0, v0}, Landroid/view/ViewGroup$Injector;->clearChildFocus(Landroid/view/ViewGroup;Landroid/view/View;)V
+ 
      return-void
  .end method
  
  .method protected detachViewsFromParent(II)V
!     .locals 1
      .parameter "start"
      .parameter "count"
  
      .prologue
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup;->removeFromArray(II)V
  
+     iget-object v0, p0, Landroid/view/ViewGroup;->mFocused:Landroid/view/View;
+ 
+     invoke-static {p0, v0}, Landroid/view/ViewGroup$Injector;->clearChildFocus(Landroid/view/ViewGroup;Landroid/view/View;)V
+ 
      return-void
  .end method
  
***************
*** 11088,11093 ****
--- 11125,11132 ----
      invoke-virtual {p0, p1}, Landroid/view/ViewGroup;->notifyChildOfDrag(Landroid/view/View;)Z
  
      :cond_1
+     invoke-static {p0}, Landroid/view/ViewGroup$Injector;->tagChildSequenceState(Landroid/view/ViewGroup;)V
+ 
      return-void
  
      :cond_2
***************
*** 12846,12851 ****
--- 12885,12900 ----
  
      return-void
  .end method
+ 
+ .method public setChildSequenceStateTaggingListener(Landroid/view/ViewGroup$ChildSequenceStateTaggingListener;)V
+     .locals 0
+     .parameter "listener"
+ 
+     .prologue
+     iput-object p1, p0, Landroid/view/ViewGroup;->mChildSequenceStateTaggingListener:Landroid/view/ViewGroup$ChildSequenceStateTaggingListener;
+ 
+     return-void
+ .end method
  
  .method protected setChildrenDrawingCacheEnabled(Z)V
      .locals 5
