*** ./smali/android/view/WindowManager$LayoutParams.smali	2013-11-18 12:21:49.662342905 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/view/WindowManager$LayoutParams.smali	2013-11-18 12:18:52.906339587 +0800
***************
*** 295,300 ****
--- 295,302 ----
  
  .field public dimAmount:F
  
+ .field public extraFlags:I
+ 
  .field public flags:I
      .annotation runtime Landroid/view/ViewDebug$ExportedProperty;
          flagMapping = {
***************
*** 1031,1036 ****
--- 1033,1044 ----
  
      move-result v0
  
+     iput v0, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+     
+     invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v0
+ 
      iput v0, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
***************
*** 1387,1392 ****
--- 1395,1413 ----
      or-int/lit8 v0, v0, 0x4
  
      :cond_9
+     iget v1, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     iget v2, p1, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     if-eq v1, v2, :cond_miui_0
+         
+     iget v1, p1, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     iput v1, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     or-int/lit8 v0, v0, 0x4
+     
+     :cond_miui_0
      iget v1, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      iget v2, p1, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
***************
*** 2017,2022 ****
--- 2038,2055 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     const-string v1, " extfl=#"
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     iget v1, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     invoke-static {v1}, Ljava/lang/Integer;->toHexString(I)Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
      iget v1, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      if-eqz v1, :cond_4
***************
*** 2300,2305 ****
--- 2333,2342 ----
      iget v0, p0, Landroid/view/WindowManager$LayoutParams;->flags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
+     
+     iget v0, p0, Landroid/view/WindowManager$LayoutParams;->extraFlags:I
+ 
+     invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
      iget v0, p0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
