*** ./smali/android/view/WindowManagerPolicy.smali	2013-11-08 15:41:22.224921507 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/view/WindowManagerPolicy.smali	2013-11-08 15:38:27.544921528 +0800
***************
*** 22,27 ****
--- 22,29 ----
  
  .field public static final ACTION_PASS_TO_USER:I = 0x1
  
+ .field public static final ACTION_POKE_USER_ACTIVITY:I = 0x2
+ 
  .field public static final ACTION_WAKE_UP:I = 0x2
  
  .field public static final EXTRA_HDMI_PLUGGED_STATE:Ljava/lang/String; = "state"
