*** ./smali/com/android/server/am/ActivityManagerService.smali	2013-11-18 12:21:24.362342430 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/services.jar.out/./smali/com/android/server/am/ActivityManagerService.smali	2013-11-18 12:18:28.530339129 +0800
***************
*** 23,29 ****
          Lcom/android/server/am/ActivityManagerService$ProcessChangeItem;,
          Lcom/android/server/am/ActivityManagerService$Identity;,
          Lcom/android/server/am/ActivityManagerService$ForegroundToken;,
!         Lcom/android/server/am/ActivityManagerService$PendingActivityLaunch;
      }
  .end annotation
  
--- 23,30 ----
          Lcom/android/server/am/ActivityManagerService$ProcessChangeItem;,
          Lcom/android/server/am/ActivityManagerService$Identity;,
          Lcom/android/server/am/ActivityManagerService$ForegroundToken;,
!         Lcom/android/server/am/ActivityManagerService$PendingActivityLaunch;,
!         Lcom/android/server/am/ActivityManagerService$Injector;
      }
  .end annotation
  
***************
*** 1350,1355 ****
--- 1351,1358 ----
      .local v1, systemDir:Ljava/io/File;
      invoke-virtual {v1}, Ljava/io/File;->mkdirs()Z
  
+     invoke-static {}, Lcom/android/server/am/ExtraActivityManagerService;->init()V
+ 
      new-instance v2, Lcom/android/server/am/BatteryStatsService;
  
      new-instance v3, Ljava/io/File;
***************
*** 4991,4996 ****
--- 4994,5015 ----
      move-result-object v9
  
      .restart local v9       #queue:Lcom/android/server/am/BroadcastQueue;
+     move-object/from16 v0, p0
+ 
+     iget-object v4, v0, Lcom/android/server/am/ActivityManagerService;->mContext:Landroid/content/Context;
+ 
+     invoke-virtual/range {p0 .. p0}, Lcom/android/server/am/ActivityManagerService;->getRunningAppProcesses()Ljava/util/List;
+ 
+     move-result-object v6
+ 
+     invoke-virtual/range {v46 .. v46}, Landroid/content/Intent;->getAction()Ljava/lang/String;
+ 
+     move-result-object v10
+ 
+     move-object/from16 v0, v25
+ 
+     invoke-static {v4, v0, v6, v10}, Lcom/android/server/am/ExtraActivityManagerService;->adjustMediaButtonReceivers(Landroid/content/Context;Ljava/util/List;Ljava/util/List;Ljava/lang/String;)V
+ 
      new-instance v8, Lcom/android/server/am/BroadcastRecord;
  
      const/16 v32, 0x0
***************
*** 7833,7843 ****
      const/16 v25, 0x0
  
      .local v25, interesting:Z
!     move-object/from16 v0, p1
  
!     move-object/from16 v1, p5
  
!     if-ne v0, v1, :cond_1a
  
      const/4 v13, 0x0
  
--- 7852,7868 ----
      const/16 v25, 0x0
  
      .local v25, interesting:Z
!     move-object/from16 v0, p0
  
!     move-object/from16 v1, p1
  
!     move-object/from16 v2, p5
! 
!     invoke-static {v0, v1, v2}, Lcom/android/server/am/ActivityManagerService$Injector;->isForegroundApp(Lcom/android/server/am/ActivityManagerService;Lcom/android/server/am/ProcessRecord;Lcom/android/server/am/ProcessRecord;)Z
! 
!     move-result v3
! 
!     if-eqz v3, :cond_1a
  
      const/4 v13, 0x0
  
***************
*** 10352,10357 ****
--- 10377,10388 ----
  
      invoke-virtual {v12, v3, v0}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     const-string v3, "crash"
+ 
+     move-object/from16 v0, p2
+ 
+     invoke-virtual {v12, v3, v0}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+ 
      iput-object v12, v15, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      move-object/from16 v0, p0
***************
*** 19109,19115 ****
      move-result-object v1
  
      .local v1, context:Landroid/content/Context;
!     const v4, 0x103006b
  
      invoke-virtual {v1, v4}, Landroid/content/Context;->setTheme(I)V
  
--- 19140,19146 ----
      move-result-object v1
  
      .local v1, context:Landroid/content/Context;
!     const v4, 0x60d003a
  
      invoke-virtual {v1, v4}, Landroid/content/Context;->setTheme(I)V
  
***************
*** 26853,26858 ****
--- 26884,26900 ----
      throw v0
  
      :cond_0
+     invoke-direct {p0, p3, p4, p7}, Lcom/android/server/am/ActivityManagerService;->checkRunningCompatibility(Landroid/content/Intent;Ljava/lang/String;I)Z
+ 
+     move-result v6
+ 
+     if-nez v6, :cond_miui_0
+ 
+     const/4 v6, -0x1
+ 
+     return v6
+ 
+     :cond_miui_0
      monitor-enter p0
  
      :try_start_0
***************
*** 39823,39838 ****
      .end local v23           #uss:Lcom/android/server/am/UserStartedState;
      :cond_2
      monitor-exit p0
  
      return-void
  
      .end local v17           #NP:I
      :catchall_0
      move-exception v2
  
      monitor-exit p0
!     :try_end_0
!     .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      throw v2
  .end method
--- 39865,39898 ----
      .end local v23           #uss:Lcom/android/server/am/UserStartedState;
      :cond_2
      monitor-exit p0
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     move-object/from16 v0, p0
+ 
+     iget v2, v0, Lcom/android/server/am/ActivityManagerService;->mFactoryTest:I
+ 
+     const/4 v3, 0x1
+ 
+     if-eq v2, v3, :cond_miui_0
  
+     move-object/from16 v0, p0
+ 
+     iget-object v2, v0, Lcom/android/server/am/ActivityManagerService;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v2}, Lcom/android/server/am/ExtraActivityManagerService;->finishBooting(Landroid/content/Context;)V
+ 
+     :cond_miui_0
      return-void
  
      .end local v17           #NP:I
      :catchall_0
      move-exception v2
  
+     :try_start_miui_0
      monitor-exit p0
!     :try_end_miui_0
!     .catchall {:try_start_miui_0 .. :try_end_miui_0} :catchall_0
  
      throw v2
  .end method
***************
*** 41425,41431 ****
      return-object v1
  
      :cond_0
!     const/4 v1, 0x0
  
      goto :goto_0
  
--- 41485,41493 ----
      return-object v1
  
      :cond_0
!     invoke-static {p0, p1}, Lcom/android/server/am/ActivityManagerService$Injector;->getCallingUidPackage(Lcom/android/server/am/ActivityManagerService;Landroid/os/IBinder;)Ljava/lang/String;
! 
!     move-result-object v1
  
      goto :goto_0
  
***************
*** 57421,57426 ****
--- 57483,57499 ----
      throw v0
  
      :cond_0
+     invoke-direct {p0, p2, p3}, Lcom/android/server/am/ActivityManagerService;->checkRunningCompatibility(Landroid/content/Intent;Ljava/lang/String;)Z
+ 
+     move-result v8
+ 
+     if-nez v8, :cond_miui_0
+ 
+     const/4 v8, 0x0
+ 
+     return-object v8
+ 
+     :cond_miui_0
      monitor-enter p0
  
      :try_start_0
***************
*** 61749,61754 ****
--- 61822,61841 ----
  
      invoke-direct/range {v2 .. v16}, Lcom/android/server/am/ActivityManagerService;->broadcastIntentLocked(Lcom/android/server/am/ProcessRecord;Ljava/lang/String;Landroid/content/Intent;Ljava/lang/String;Landroid/content/IIntentReceiver;ILjava/lang/String;Landroid/os/Bundle;Ljava/lang/String;ZZIII)I
  
+     move-object/from16 v0, p0
+ 
+     iget-object v2, v0, Lcom/android/server/am/ActivityManagerService;->mContext:Landroid/content/Context;
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v3, v0, Lcom/android/server/am/ActivityManagerService;->mHandler:Landroid/os/Handler;
+ 
+     move/from16 v0, v19
+ 
+     move-object/from16 v1, v24
+ 
+     invoke-static {v0, v1, v2, v3}, Landroid/app/MiuiThemeHelper;->handleExtraConfigurationChanges(ILandroid/content/res/Configuration;Landroid/content/Context;Landroid/os/Handler;)V
+ 
      and-int/lit8 v2, v19, 0x4
  
      if-eqz v2, :cond_9
***************
*** 64564,64566 ****
--- 64651,64720 ----
  
      goto :goto_1
  .end method
+ 
+ .method private checkRunningCompatibility(Landroid/content/Intent;Ljava/lang/String;)Z
+     .locals 2
+     .parameter "service"
+     .parameter "resolvedType"
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/server/am/ActivityManagerService;->mSystemReady:Z
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/server/am/ActivityManagerService;->mContext:Landroid/content/Context;
+ 
+     invoke-static {}, Landroid/os/Binder;->getCallingUid()I
+ 
+     move-result v1
+ 
+     invoke-static {v1}, Landroid/os/UserId;->getUserId(I)I
+ 
+     move-result v1
+ 
+     invoke-static {v0, p1, p2, v1}, Lcom/android/server/am/ExtraActivityManagerService;->checkRunningCompatibility(Landroid/content/Context;Landroid/content/Intent;Ljava/lang/String;I)Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_0
+ 
+     const/4 v0, 0x0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x1
+ 
+     goto :goto_0
+ .end method
+ 
+ .method private checkRunningCompatibility(Landroid/content/Intent;Ljava/lang/String;I)Z
+     .locals 1
+     .parameter "service"
+     .parameter "resolvedType"
+     .parameter "userId"
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/server/am/ActivityManagerService;->mSystemReady:Z
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/server/am/ActivityManagerService;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v0, p1, p2, p3}, Lcom/android/server/am/ExtraActivityManagerService;->checkRunningCompatibility(Landroid/content/Context;Landroid/content/Intent;Ljava/lang/String;I)Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_0
+ 
+     const/4 v0, 0x0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x1
+ 
+     goto :goto_0
+ .end method
