*** ./smali/android/hardware/SensorManager.smali	2013-11-18 12:21:59.834343096 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/hardware/SensorManager.smali	2013-11-18 12:19:03.170339779 +0800
***************
*** 2294,2299 ****
--- 2294,2309 ----
  
  
  # virtual methods
+ .method public calibrateSensor(Landroid/hardware/Sensor;)Z
+     .locals 1
+     .parameter "sensor"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
  .method public getDefaultSensor(I)Landroid/hardware/Sensor;
      .locals 2
      .parameter "type"
