*** ./smali/com/android/internal/telephony/BaseCommands.smali	2013-11-18 12:22:18.438343445 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/BaseCommands.smali	2013-11-18 12:19:21.074340114 +0800
***************
*** 313,318 ****
--- 313,329 ----
      return-void
  .end method
  
+ .method public static getLteOnCdmaModeStatic()I
+     .locals 1
+ 
+     .prologue
+     invoke-static {}, Landroid/telephony/TelephonyManager;->getLteOnCdmaModeStatic()I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  
  # virtual methods
  .method public getLteOnCdmaMode()I
