*** ./smali/com/android/internal/telephony/DataConnectionTracker.smali	2013-11-10 11:52:30.325671429 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/DataConnectionTracker.smali	2013-11-10 11:53:50.293673477 +0800
***************
*** 9,15 ****
          Lcom/android/internal/telephony/DataConnectionTracker$3;,
          Lcom/android/internal/telephony/DataConnectionTracker$RecoveryAction;,
          Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;,
!         Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;
      }
  .end annotation
  
--- 9,16 ----
          Lcom/android/internal/telephony/DataConnectionTracker$3;,
          Lcom/android/internal/telephony/DataConnectionTracker$RecoveryAction;,
          Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;,
!         Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;,
!         Lcom/android/internal/telephony/DataConnectionTracker$Injector;
      }
  .end annotation
  
***************
*** 2808,2813 ****
--- 2809,2818 ----
  
      .local v0, result:Z
      :goto_0
+     invoke-static {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker$Injector;->getAnyDataEnabled(Lcom/android/internal/telephony/DataConnectionTracker;Z)Z
+ 
+     move-result v0
+ 
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 3840,3845 ****
--- 3845,3893 ----
      throw v1
  .end method
  
+ .method protected isMmsDataEnabled()Z
+     .locals 3
+ 
+     .prologue
+     const/4 v0, 0x1
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
+ 
+     const-string v2, "mms"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
+ 
+     invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
+ 
+     move-result-object v1
+ 
+     const-string v2, "always_enable_mms"
+ 
+     invoke-static {v1, v2, v0}, Landroid/provider/Settings$System;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
+ 
+     move-result v1
+ 
+     if-ne v1, v0, :cond_0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method protected abstract log(Ljava/lang/String;)V
  .end method
  
