*** ./smali/com/android/internal/telephony/SmsUsageMonitor.smali	2013-11-08 15:41:51.792921502 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/SmsUsageMonitor.smali	2013-11-08 15:38:59.148921524 +0800
***************
*** 1679,1684 ****
--- 1679,1697 ----
      .parameter "smsWaiting"
  
      .prologue
+     const-string v1, "com.android.mms"
+ 
+     invoke-virtual {v1, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_miui_0
+ 
+     const/4 v1, 0x1
+ 
+     return v1
+ 
+     :cond_miui_0
      iget-object v2, p0, Lcom/android/internal/telephony/SmsUsageMonitor;->mSmsStamp:Ljava/util/HashMap;
  
      monitor-enter v2
