*** ./smali/android/app/Activity.smali	2013-11-08 15:41:38.116921505 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/app/Activity.smali	2013-11-08 15:38:45.244921526 +0800
***************
*** 15,21 ****
      value = {
          Landroid/app/Activity$ManagedCursor;,
          Landroid/app/Activity$NonConfigurationInstances;,
!         Landroid/app/Activity$ManagedDialog;
      }
  .end annotation
  
--- 15,22 ----
      value = {
          Landroid/app/Activity$ManagedCursor;,
          Landroid/app/Activity$NonConfigurationInstances;,
!         Landroid/app/Activity$ManagedDialog;,
!         Landroid/app/Activity$Injector;
      }
  .end annotation
  
***************
*** 439,447 ****
      return-void
  
      :cond_1
!     new-instance v1, Lcom/android/internal/app/ActionBarImpl;
  
!     invoke-direct {v1, p0}, Lcom/android/internal/app/ActionBarImpl;-><init>(Landroid/app/Activity;)V
  
      iput-object v1, p0, Landroid/app/Activity;->mActionBar:Lcom/android/internal/app/ActionBarImpl;
  
--- 440,448 ----
      return-void
  
      :cond_1
!     invoke-static {p0}, Landroid/app/Activity$Injector;->generateActionBar(Landroid/app/Activity;)Lcom/android/internal/app/ActionBarImpl;
  
!     move-result-object v1
  
      iput-object v1, p0, Landroid/app/Activity;->mActionBar:Lcom/android/internal/app/ActionBarImpl;
  
***************
*** 2432,2437 ****
--- 2433,2464 ----
      goto :goto_0
  .end method
  
+ .method public getMiuiActionBar()Lmiui/v5/app/MiuiActionBar;
+     .locals 2
+ 
+     .prologue
+     invoke-virtual {p0}, Landroid/app/Activity;->getActionBar()Landroid/app/ActionBar;
+ 
+     move-result-object v0
+ 
+     .local v0, bar:Landroid/app/ActionBar;
+     instance-of v1, v0, Lmiui/v5/app/MiuiActionBar;
+ 
+     if-eqz v1, :cond_0
+ 
+     check-cast v0, Lmiui/v5/app/MiuiActionBar;
+ 
+     .end local v0           #bar:Landroid/app/ActionBar;
+     :goto_0
+     return-object v0
+ 
+     .restart local v0       #bar:Landroid/app/ActionBar;
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public final getParent()Landroid/app/Activity;
      .locals 1
  
***************
*** 2705,2710 ****
--- 2732,2746 ----
      return v0
  .end method
  
+ .method getToken()Landroid/os/IBinder;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/app/Activity;->mToken:Landroid/os/IBinder;
+ 
+     return-object v0
+ .end method
+ 
  .method public final getVolumeControlStream()I
      .locals 1
  
***************
*** 3474,3479 ****
--- 3510,3517 ----
  
      iput-boolean v3, p0, Landroid/app/Activity;->mCalled:Z
  
+     invoke-static {p0}, Landroid/app/Activity$Injector;->setActivityGravity(Landroid/app/Activity;)V
+ 
      return-void
  
      :cond_3
***************
*** 4786,4792 ****
      return v0
  
      :cond_0
!     const/4 v0, 0x0
  
      goto :goto_0
  .end method
--- 4824,4832 ----
      return v0
  
      :cond_0
!     invoke-static {p0, p1}, Landroid/app/Activity$Injector;->onOptionsItemSelected(Landroid/app/Activity;Landroid/view/MenuItem;)Z
! 
!     move-result v0
  
      goto :goto_0
  .end method
***************
*** 5077,5082 ****
--- 5117,5124 ----
  
      iput-boolean v0, p0, Landroid/app/Activity;->mCalled:Z
  
+     invoke-static {p0}, Landroid/app/Activity$Injector;->checkAccessControl(Landroid/app/Activity;)V
+ 
      return-void
  .end method
  
***************
*** 5529,5534 ****
--- 5571,5578 ----
      .prologue
      const/4 v2, 0x0
  
+     invoke-static {p0}, Landroid/app/Activity$Injector;->onWindowHide(Landroid/app/Activity;)V
+ 
      iget-object v0, p0, Landroid/app/Activity;->mFragments:Landroid/app/FragmentManagerImpl;
  
      invoke-virtual {v0}, Landroid/app/FragmentManagerImpl;->dispatchPause()V
***************
*** 5941,5946 ****
--- 5985,5992 ----
      throw v0
  
      :cond_1
+     invoke-static {p0}, Landroid/app/Activity$Injector;->onWindowShow(Landroid/app/Activity;)V
+ 
      return-void
  .end method
  
