*** ./smali/android/app/ContextImpl$13.smali	2013-11-10 11:53:11.977672496 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/android/app/ContextImpl$13.smali	2013-11-10 11:54:31.389674529 +0800
***************
*** 31,37 ****
      .parameter "ctx"
  
      .prologue
!     new-instance v0, Landroid/app/DownloadManager;
  
      invoke-virtual {p1}, Landroid/app/ContextImpl;->getContentResolver()Landroid/content/ContentResolver;
  
--- 31,37 ----
      .parameter "ctx"
  
      .prologue
!     new-instance v0, Landroid/app/MiuiDownloadManager;
  
      invoke-virtual {p1}, Landroid/app/ContextImpl;->getContentResolver()Landroid/content/ContentResolver;
  
***************
*** 41,47 ****
  
      move-result-object v2
  
!     invoke-direct {v0, v1, v2}, Landroid/app/DownloadManager;-><init>(Landroid/content/ContentResolver;Ljava/lang/String;)V
  
      return-object v0
  .end method
--- 41,47 ----
  
      move-result-object v2
  
!     invoke-direct {v0, v1, v2}, Landroid/app/MiuiDownloadManager;-><init>(Landroid/content/ContentResolver;Ljava/lang/String;)V
  
      return-object v0
  .end method
