*** ./smali/android/app/INotificationManager.smali	2013-11-08 15:41:37.816921504 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/app/INotificationManager.smali	2013-11-08 15:38:44.948921526 +0800
***************
*** 15,20 ****
--- 15,28 ----
  
  
  # virtual methods
+ .method public abstract areNotificationsEnabled(Ljava/lang/String;)Z
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ .end method
+ 
  .method public abstract areNotificationsEnabledForPackage(Ljava/lang/String;)Z
      .annotation system Ldalvik/annotation/Throws;
          value = {
