*** ./smali/android/app/INotificationManager.smali	2013-11-10 11:53:09.837672441 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/android/app/INotificationManager.smali	2013-11-10 11:54:29.253674474 +0800
***************
*** 15,20 ****
--- 15,28 ----
  
  
  # virtual methods
+ .method public abstract areNotificationsEnabled(Ljava/lang/String;)Z
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ .end method
+ 
  .method public abstract areNotificationsEnabledForPackage(Ljava/lang/String;)Z
      .annotation system Ldalvik/annotation/Throws;
          value = {
