*** ./smali/android/widget/Chronometer.smali	2013-11-18 12:21:57.890343058 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/widget/Chronometer.smali	2013-11-18 12:19:01.082339740 +0800
***************
*** 9,15 ****
  
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/widget/Chronometer$OnChronometerTickListener;
      }
  .end annotation
  
--- 9,16 ----
  
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/widget/Chronometer$OnChronometerTickListener;,
!         Landroid/widget/Chronometer$Injector;
      }
  .end annotation
  
***************
*** 247,255 ****
  
      div-long/2addr v2, v5
  
!     iget-object v5, p0, Landroid/widget/Chronometer;->mRecycle:Ljava/lang/StringBuilder;
  
!     invoke-static {v5, v2, v3}, Landroid/text/format/DateUtils;->formatElapsedTime(Ljava/lang/StringBuilder;J)Ljava/lang/String;
  
      move-result-object v4
  
--- 248,258 ----
  
      div-long/2addr v2, v5
  
!     iget-object v5, p0, Landroid/widget/Chronometer;->mContext:Landroid/content/Context;
  
!     iget-object v6, p0, Landroid/widget/Chronometer;->mRecycle:Ljava/lang/StringBuilder;
! 
!     invoke-static {v5, v6, v2, v3}, Landroid/widget/Chronometer$Injector;->formatElapsedTime(Landroid/content/Context;Ljava/lang/StringBuilder;J)Ljava/lang/String;
  
      move-result-object v4
  
