*** ./smali/android/widget/EdgeEffect.smali	2013-11-18 12:21:55.698343018 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/widget/EdgeEffect.smali	2013-11-18 12:18:58.866339698 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "EdgeEffect.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Landroid/widget/EdgeEffect$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field private static final EPSILON:F = 0.0010f
  
***************
*** 136,152 ****
      move-result-object v0
  
      .local v0, res:Landroid/content/res/Resources;
!     const v1, 0x1080436
! 
!     invoke-virtual {v0, v1}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/widget/EdgeEffect;->mEdge:Landroid/graphics/drawable/Drawable;
  
!     const v1, 0x1080437
! 
!     invoke-virtual {v0, v1}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
--- 144,156 ----
      move-result-object v0
  
      .local v0, res:Landroid/content/res/Resources;
!     invoke-static {p1}, Landroid/widget/EdgeEffect$Injector;->getOverScrollEdge(Landroid/content/Context;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/widget/EdgeEffect;->mEdge:Landroid/graphics/drawable/Drawable;
  
!     invoke-static {p1}, Landroid/widget/EdgeEffect$Injector;->getOverScrollGlow(Landroid/content/Context;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
