*** ./smali/android/widget/NumberPicker.smali	2013-11-08 15:41:29.576921506 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/widget/NumberPicker.smali	2013-11-08 15:38:35.228921527 +0800
***************
*** 16,22 ****
          Landroid/widget/NumberPicker$Formatter;,
          Landroid/widget/NumberPicker$OnScrollListener;,
          Landroid/widget/NumberPicker$OnValueChangeListener;,
!         Landroid/widget/NumberPicker$TwoDigitFormatter;
      }
  .end annotation
  
--- 16,23 ----
          Landroid/widget/NumberPicker$Formatter;,
          Landroid/widget/NumberPicker$OnScrollListener;,
          Landroid/widget/NumberPicker$OnValueChangeListener;,
!         Landroid/widget/NumberPicker$TwoDigitFormatter;,
!         Landroid/widget/NumberPicker$Injector;
      }
  .end annotation
  
***************
*** 32,47 ****
  
  .field private static final SELECTOR_MAX_FLING_VELOCITY_ADJUSTMENT:I = 0x8
  
- .field private static final SELECTOR_MIDDLE_ITEM_INDEX:I = 0x1
- 
- .field private static final SELECTOR_WHEEL_ITEM_COUNT:I = 0x3
- 
  .field private static final SIZE_UNSPECIFIED:I = -0x1
  
  .field private static final SNAP_SCROLL_DURATION:I = 0x12c
  
  .field private static final TOP_AND_BOTTOM_FADING_EDGE_STRENGTH:F = 0.9f
  
  .field private static final UNSCALED_DEFAULT_SELECTION_DIVIDERS_DISTANCE:I = 0x30
  
  .field private static final UNSCALED_DEFAULT_SELECTION_DIVIDER_HEIGHT:I = 0x2
--- 33,46 ----
  
  .field private static final SELECTOR_MAX_FLING_VELOCITY_ADJUSTMENT:I = 0x8
  
  .field private static final SIZE_UNSPECIFIED:I = -0x1
  
  .field private static final SNAP_SCROLL_DURATION:I = 0x12c
  
  .field private static final TOP_AND_BOTTOM_FADING_EDGE_STRENGTH:F = 0.9f
  
+ .field public static final TWO_DIGIT_FORMATTER:Landroid/widget/NumberPicker$Formatter; = null
+ 
  .field private static final UNSCALED_DEFAULT_SELECTION_DIVIDERS_DISTANCE:I = 0x30
  
  .field private static final UNSCALED_DEFAULT_SELECTION_DIVIDER_HEIGHT:I = 0x2
***************
*** 50,55 ****
--- 49,58 ----
  
  
  # instance fields
+ .field private SELECTOR_MIDDLE_ITEM_INDEX:I
+ 
+ .field private SELECTOR_WHEEL_ITEM_COUNT:I
+ 
  .field private mAccessibilityNodeProvider:Landroid/widget/NumberPicker$AccessibilityNodeProviderImpl;
  
  .field private final mAdjustScroller:Landroid/widget/Scroller;
***************
*** 86,91 ****
--- 89,96 ----
  
  .field private final mInputText:Landroid/widget/EditText;
  
+ .field private mIsV5:Z
+ 
  .field private mLastDownEventTime:J
  
  .field private mLastDownEventY:F
***************
*** 96,102 ****
  
  .field private mLongPressUpdateInterval:J
  
! .field private final mMaxHeight:I
  
  .field private mMaxValue:I
  
--- 101,107 ----
  
  .field private mLongPressUpdateInterval:J
  
! .field private mMaxHeight:I
  
  .field private mMaxValue:I
  
***************
*** 112,117 ****
--- 117,124 ----
  
  .field private mMinimumFlingVelocity:I
  
+ .field private mNeedSoftInput:Z
+ 
  .field private mOnScrollListener:Landroid/widget/NumberPicker$OnScrollListener;
  
  .field private mOnValueChangeListener:Landroid/widget/NumberPicker$OnValueChangeListener;
***************
*** 126,132 ****
  
  .field private final mSelectionDividerHeight:I
  
! .field private final mSelectionDividersDistance:I
  
  .field private mSelectorElementHeight:I
  
--- 133,139 ----
  
  .field private final mSelectionDividerHeight:I
  
! .field private mSelectionDividersDistance:I
  
  .field private mSelectorElementHeight:I
  
***************
*** 141,147 ****
      .end annotation
  .end field
  
! .field private final mSelectorIndices:[I
  
  .field private mSelectorTextGapHeight:I
  
--- 148,154 ----
      .end annotation
  .end field
  
! .field private mSelectorIndices:[I
  
  .field private mSelectorTextGapHeight:I
  
***************
*** 179,184 ****
--- 186,195 ----
  
      sput-object v0, Landroid/widget/NumberPicker;->sTwoDigitFormatter:Landroid/widget/NumberPicker$TwoDigitFormatter;
  
+     sget-object v0, Landroid/widget/NumberPicker;->sTwoDigitFormatter:Landroid/widget/NumberPicker$TwoDigitFormatter;
+ 
+     sput-object v0, Landroid/widget/NumberPicker;->TWO_DIGIT_FORMATTER:Landroid/widget/NumberPicker$Formatter;
+ 
      const/16 v0, 0x1e
  
      new-array v0, v0, [C
***************
*** 260,265 ****
--- 271,298 ----
      .prologue
      invoke-direct/range {p0 .. p3}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     const/4 v14, 0x3
+ 
+     move-object/from16 v0, p0
+ 
+     iput v14, v0, Landroid/widget/NumberPicker;->SELECTOR_WHEEL_ITEM_COUNT:I
+ 
+     move-object/from16 v0, p0
+ 
+     iget v14, v0, Landroid/widget/NumberPicker;->SELECTOR_WHEEL_ITEM_COUNT:I
+ 
+     div-int/lit8 v14, v14, 0x2
+ 
+     move-object/from16 v0, p0
+ 
+     iput v14, v0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
+ 
+     const/4 v14, 0x1
+ 
+     move-object/from16 v0, p0
+ 
+     iput-boolean v14, v0, Landroid/widget/NumberPicker;->mNeedSoftInput:Z
+ 
      const-wide/16 v14, 0x12c
  
      move-object/from16 v0, p0
***************
*** 274,280 ****
  
      iput-object v14, v0, Landroid/widget/NumberPicker;->mSelectorIndexToStringCache:Landroid/util/SparseArray;
  
!     const/4 v14, 0x3
  
      new-array v14, v14, [I
  
--- 307,315 ----
  
      iput-object v14, v0, Landroid/widget/NumberPicker;->mSelectorIndexToStringCache:Landroid/util/SparseArray;
  
!     move-object/from16 v0, p0
! 
!     iget v14, v0, Landroid/widget/NumberPicker;->SELECTOR_WHEEL_ITEM_COUNT:I
  
      new-array v14, v14, [I
  
***************
*** 294,299 ****
--- 329,340 ----
  
      iput v14, v0, Landroid/widget/NumberPicker;->mScrollState:I
  
+     const/4 v14, 0x0
+ 
+     move-object/from16 v0, p0
+ 
+     iput-boolean v14, v0, Landroid/widget/NumberPicker;->mIsV5:Z
+ 
      sget-object v14, Lcom/android/internal/R$styleable;->NumberPicker:[I
  
      const/4 v15, 0x0
***************
*** 2377,2383 ****
      .local v0, editTextTextPosition:I
      iget v5, p0, Landroid/widget/NumberPicker;->mSelectorElementHeight:I
  
!     mul-int/lit8 v5, v5, 0x1
  
      sub-int v5, v0, v5
  
--- 2418,2426 ----
      .local v0, editTextTextPosition:I
      iget v5, p0, Landroid/widget/NumberPicker;->mSelectorElementHeight:I
  
!     iget v6, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     mul-int/2addr v5, v6
  
      sub-int v5, v0, v5
  
***************
*** 2418,2424 ****
  
      if-ge v1, v4, :cond_1
  
!     add-int/lit8 v4, v1, -0x1
  
      add-int v2, v0, v4
  
--- 2461,2469 ----
  
      if-ge v1, v4, :cond_1
  
!     iget v4, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     sub-int v4, v1, v4
  
      add-int v2, v0, v4
  
***************
*** 2963,2968 ****
--- 3008,3020 ----
      .prologue
      const/4 v2, 0x0
  
+     iget-boolean v1, p0, Landroid/widget/NumberPicker;->mNeedSoftInput:Z
+ 
+     if-nez v1, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      invoke-static {}, Landroid/view/inputmethod/InputMethodManager;->peekInstance()Landroid/view/inputmethod/InputMethodManager;
  
      move-result-object v0
***************
*** 3062,3067 ****
--- 3114,3127 ----
      goto :goto_2
  
      :cond_4
+     iget-object v8, p0, Landroid/widget/NumberPicker;->mFormatter:Landroid/widget/NumberPicker$Formatter;
+ 
+     iget v9, p0, Landroid/widget/NumberPicker;->mMaxValue:I
+ 
+     invoke-static {v8, v9, v5}, Landroid/widget/NumberPicker$Injector;->getNumberCount(Landroid/widget/NumberPicker$Formatter;II)I
+ 
+     move-result v5
+ 
      int-to-float v8, v5
  
      mul-float/2addr v8, v3
***************
*** 3265,3270 ****
--- 3325,3339 ----
  
  
  # virtual methods
+ .method protected adjustScrollWheel()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Landroid/widget/NumberPicker;->ensureScrollWheelAdjusted()Z
+ 
+     return-void
+ .end method
+ 
  .method public computeScroll()V
      .locals 4
  
***************
*** 3619,3624 ****
--- 3688,3720 ----
      return v0
  .end method
  
+ .method protected getBottomSelectionDividerBottom()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mBottomSelectionDividerBottom:I
+ 
+     return v0
+ .end method
+ 
+ .method protected getCurrentScrollOffset()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mCurrentScrollOffset:I
+ 
+     return v0
+ .end method
+ 
+ .method protected getDecrementVirtualBUttonPressed()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/widget/NumberPicker;->mDecrementVirtualButtonPressed:Z
+ 
+     return v0
+ .end method
+ 
  .method public getDisplayedValues()[Ljava/lang/String;
      .locals 1
  
***************
*** 3628,3633 ****
--- 3724,3756 ----
      return-object v0
  .end method
  
+ .method protected getEditText()Landroid/widget/EditText;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/NumberPicker;->mInputText:Landroid/widget/EditText;
+ 
+     return-object v0
+ .end method
+ 
+ .method protected getIncrementVirtualButtonPressed()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/widget/NumberPicker;->mIncrementVirtualButtonPressed:Z
+ 
+     return v0
+ .end method
+ 
+ .method protected getInitialScrollOffset()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mInitialScrollOffset:I
+ 
+     return v0
+ .end method
+ 
  .method public getMaxValue()I
      .locals 1
  
***************
*** 3646,3651 ****
--- 3769,3837 ----
      return v0
  .end method
  
+ .method protected getMinWidth()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mMinWidth:I
+ 
+     return v0
+ .end method
+ 
+ .method protected getScrollState()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mScrollState:I
+ 
+     return v0
+ .end method
+ 
+ .method protected getSelectorElementHeight()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mSelectorElementHeight:I
+ 
+     return v0
+ .end method
+ 
+ .method protected getSelectorIndexToStringCache()Landroid/util/SparseArray;
+     .locals 1
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "()",
+             "Landroid/util/SparseArray",
+             "<",
+             "Ljava/lang/String;",
+             ">;"
+         }
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/NumberPicker;->mSelectorIndexToStringCache:Landroid/util/SparseArray;
+ 
+     return-object v0
+ .end method
+ 
+ .method protected getSelectorIndices()[I
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/NumberPicker;->mSelectorIndices:[I
+ 
+     return-object v0
+ .end method
+ 
+ .method protected getSelectorWheelPaint()Landroid/graphics/Paint;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/NumberPicker;->mSelectorWheelPaint:Landroid/graphics/Paint;
+ 
+     return-object v0
+ .end method
+ 
  .method public getSolidColor()I
      .locals 1
  
***************
*** 3655,3660 ****
--- 3841,3855 ----
      return v0
  .end method
  
+ .method protected getTextSize()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mTextSize:I
+ 
+     return v0
+ .end method
+ 
  .method protected getTopFadingEdgeStrength()F
      .locals 1
  
***************
*** 3664,3669 ****
--- 3859,3873 ----
      return v0
  .end method
  
+ .method protected getTopSelectionDividerTop()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Landroid/widget/NumberPicker;->mTopSelectionDividerTop:I
+ 
+     return v0
+ .end method
+ 
  .method public getValue()I
      .locals 1
  
***************
*** 3673,3678 ****
--- 3877,3891 ----
      return v0
  .end method
  
+ .method protected getVirtualButtonPressedDrawable()Landroid/graphics/drawable/Drawable;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/NumberPicker;->mVirtualButtonPressedDrawable:Landroid/graphics/drawable/Drawable;
+ 
+     return-object v0
+ .end method
+ 
  .method public getWrapSelectorWheel()Z
      .locals 1
  
***************
*** 3682,3687 ****
--- 3895,3909 ----
      return v0
  .end method
  
+ .method protected hasSelectorWheel()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/widget/NumberPicker;->mHasSelectorWheel:Z
+ 
+     return v0
+ .end method
+ 
  .method protected onDetachedFromWindow()V
      .locals 0
  
***************
*** 3854,3860 ****
      check-cast v4, Ljava/lang/String;
  
      .local v4, scrollSelectorValue:Ljava/lang/String;
!     const/4 v11, 0x1
  
      if-ne v3, v11, :cond_4
  
--- 4076,4084 ----
      check-cast v4, Ljava/lang/String;
  
      .local v4, scrollSelectorValue:Ljava/lang/String;
!     move-object/from16 v0, p0
! 
!     iget v11, v0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
  
      if-ne v3, v11, :cond_4
  
***************
*** 4098,4109 ****
--- 4322,4338 ----
  
      :cond_2
      :goto_1
+     iget-boolean v3, p0, Landroid/widget/NumberPicker;->mIsV5:Z
+ 
+     if-nez v3, :cond_miui_0
+ 
      invoke-virtual {p0}, Landroid/widget/NumberPicker;->getParent()Landroid/view/ViewParent;
  
      move-result-object v3
  
      invoke-interface {v3, v1}, Landroid/view/ViewParent;->requestDisallowInterceptTouchEvent(Z)V
  
+     :cond_miui_0
      iget-object v3, p0, Landroid/widget/NumberPicker;->mFlingScroller:Landroid/widget/Scroller;
  
      invoke-virtual {v3}, Landroid/widget/Scroller;->isFinished()Z
***************
*** 4618,4624 ****
  
      div-int v10, v6, v10
  
!     add-int/lit8 v8, v10, -0x1
  
      .local v8, selectorIndexOffset:I
      if-lez v8, :cond_9
--- 4847,4855 ----
  
      div-int v10, v6, v10
  
!     iget v11, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     sub-int v8, v10, v11
  
      .local v8, selectorIndexOffset:I
      if-lez v8, :cond_9
***************
*** 4665,4670 ****
--- 4896,4956 ----
      .end packed-switch
  .end method
  
+ .method protected reInitiate(IIIZ)V
+     .locals 1
+     .parameter "selectorWheelItemCount"
+     .parameter "maxHeight"
+     .parameter "selectionDividersDistance"
+     .parameter "needSoftInput"
+ 
+     .prologue
+     iput p1, p0, Landroid/widget/NumberPicker;->SELECTOR_WHEEL_ITEM_COUNT:I
+ 
+     iget v0, p0, Landroid/widget/NumberPicker;->SELECTOR_WHEEL_ITEM_COUNT:I
+ 
+     div-int/lit8 v0, v0, 0x2
+ 
+     iput v0, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
+ 
+     iget v0, p0, Landroid/widget/NumberPicker;->SELECTOR_WHEEL_ITEM_COUNT:I
+ 
+     new-array v0, v0, [I
+ 
+     iput-object v0, p0, Landroid/widget/NumberPicker;->mSelectorIndices:[I
+ 
+     iput p2, p0, Landroid/widget/NumberPicker;->mMaxHeight:I
+ 
+     iput p3, p0, Landroid/widget/NumberPicker;->mSelectionDividersDistance:I
+ 
+     iput-boolean p4, p0, Landroid/widget/NumberPicker;->mNeedSoftInput:Z
+ 
+     const/4 v0, 0x1
+ 
+     iput-boolean v0, p0, Landroid/widget/NumberPicker;->mIsV5:Z
+ 
+     return-void
+ .end method
+ 
+ .method public refreshWheel()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Landroid/widget/NumberPicker;->initializeSelectorWheelIndices()V
+ 
+     return-void
+ .end method
+ 
+ .method protected remeasureMaxWidth()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Landroid/widget/NumberPicker;->tryComputeMaxWidth()V
+ 
+     invoke-virtual {p0}, Landroid/widget/NumberPicker;->invalidate()V
+ 
+     return-void
+ .end method
+ 
  .method public scrollBy(II)V
      .locals 4
      .parameter "x"
***************
*** 4682,4688 ****
  
      if-lez p2, :cond_1
  
!     aget v1, v0, v3
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMinValue:I
  
--- 4968,4976 ----
  
      if-lez p2, :cond_1
  
!     iget v1, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     aget v1, v0, v1
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMinValue:I
  
***************
*** 4703,4709 ****
  
      if-gez p2, :cond_2
  
!     aget v1, v0, v3
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMaxValue:I
  
--- 4991,4999 ----
  
      if-gez p2, :cond_2
  
!     iget v1, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     aget v1, v0, v1
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMaxValue:I
  
***************
*** 4744,4750 ****
  
      invoke-direct {p0, v0}, Landroid/widget/NumberPicker;->decrementSelectorIndices([I)V
  
!     aget v1, v0, v3
  
      invoke-direct {p0, v1, v3}, Landroid/widget/NumberPicker;->setValueInternal(IZ)V
  
--- 5034,5042 ----
  
      invoke-direct {p0, v0}, Landroid/widget/NumberPicker;->decrementSelectorIndices([I)V
  
!     iget v1, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     aget v1, v0, v1
  
      invoke-direct {p0, v1, v3}, Landroid/widget/NumberPicker;->setValueInternal(IZ)V
  
***************
*** 4752,4758 ****
  
      if-nez v1, :cond_3
  
!     aget v1, v0, v3
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMinValue:I
  
--- 5044,5052 ----
  
      if-nez v1, :cond_3
  
!     iget v1, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     aget v1, v0, v1
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMinValue:I
  
***************
*** 4788,4794 ****
  
      invoke-direct {p0, v0}, Landroid/widget/NumberPicker;->incrementSelectorIndices([I)V
  
!     aget v1, v0, v3
  
      invoke-direct {p0, v1, v3}, Landroid/widget/NumberPicker;->setValueInternal(IZ)V
  
--- 5082,5090 ----
  
      invoke-direct {p0, v0}, Landroid/widget/NumberPicker;->incrementSelectorIndices([I)V
  
!     iget v1, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     aget v1, v0, v1
  
      invoke-direct {p0, v1, v3}, Landroid/widget/NumberPicker;->setValueInternal(IZ)V
  
***************
*** 4796,4802 ****
  
      if-nez v1, :cond_4
  
!     aget v1, v0, v3
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMaxValue:I
  
--- 5092,5100 ----
  
      if-nez v1, :cond_4
  
!     iget v1, p0, Landroid/widget/NumberPicker;->SELECTOR_MIDDLE_ITEM_INDEX:I
! 
!     aget v1, v0, v1
  
      iget v2, p0, Landroid/widget/NumberPicker;->mMaxValue:I
  
