*** ./smali/android/widget/RemoteViews.smali	2013-11-08 15:41:31.728921506 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/widget/RemoteViews.smali	2013-11-08 15:38:38.816921525 +0800
***************
*** 71,76 ****
--- 71,78 ----
  
  .field private mBitmapCache:Landroid/widget/RemoteViews$BitmapCache;
  
+ .field mDefaultTheme:I
+ 
  .field private mIsRoot:Z
  
  .field private mIsWidgetCollectionChild:Z
***************
*** 1133,1138 ****
--- 1135,1144 ----
      move-result-object v0
  
      .local v0, c:Landroid/content/Context;
+     iget v4, p0, Landroid/widget/RemoteViews;->mDefaultTheme:I
+ 
+     invoke-virtual {v0, v4}, Landroid/content/Context;->setTheme(I)V
+ 
      const-string v4, "layout_inflater"
  
      invoke-virtual {v0, v4}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 1684,1689 ****
--- 1690,1705 ----
  
      return-void
  .end method
+ 
+ .method public setDefaultTheme(I)V
+     .locals 0
+     .parameter "id"
+ 
+     .prologue
+     iput p1, p0, Landroid/widget/RemoteViews;->mDefaultTheme:I
+ 
+     return-void
+ .end method
  
  .method public setDisplayedChild(II)V
      .locals 1
