*** ./smali/android/widget/TextView.smali	2013-11-10 11:52:54.613672051 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/android/widget/TextView.smali	2013-11-10 11:54:14.697674102 +0800
***************
*** 5742,5748 ****
      goto :goto_0
  .end method
  
! .method private static isPasswordInputType(I)Z
      .locals 2
      .parameter "inputType"
  
--- 5742,5748 ----
      goto :goto_0
  .end method
  
! .method static isPasswordInputType(I)Z
      .locals 2
      .parameter "inputType"
  
***************
*** 7908,7913 ****
--- 7908,7917 ----
      packed-switch p2, :pswitch_data_0
  
      :goto_1
+     invoke-static {v0, p2, p3}, Landroid/graphics/MiuiTypeface;->getChangedTypeface(Landroid/graphics/Typeface;II)Landroid/graphics/Typeface;
+ 
+     move-result-object v0
+ 
      invoke-virtual {p0, v0, p3}, Landroid/widget/TextView;->setTypeface(Landroid/graphics/Typeface;I)V
  
      goto :goto_0
***************
*** 22007,22012 ****
--- 22011,22055 ----
      return-void
  .end method
  
+ .method public setCursorDrawableRes(I)V
+     .locals 3
+     .parameter "cursorDrawableRes"
+ 
+     .prologue
+     iput p1, p0, Landroid/widget/TextView;->mCursorDrawableRes:I
+ 
+     const/4 v0, 0x0
+ 
+     .local v0, i:I
+     :goto_0
+     iget-object v1, p0, Landroid/widget/TextView;->mEditor:Landroid/widget/Editor;
+ 
+     iget v1, v1, Landroid/widget/Editor;->mCursorCount:I
+ 
+     if-ge v0, v1, :cond_0
+ 
+     iget-object v1, p0, Landroid/widget/TextView;->mEditor:Landroid/widget/Editor;
+ 
+     iget-object v1, v1, Landroid/widget/Editor;->mCursorDrawable:[Landroid/graphics/drawable/Drawable;
+ 
+     const/4 v2, 0x0
+ 
+     aput-object v2, v1, v0
+ 
+     add-int/lit8 v0, v0, 0x1
+ 
+     goto :goto_0
+ 
+     :cond_0
+     iget-object v1, p0, Landroid/widget/TextView;->mEditor:Landroid/widget/Editor;
+ 
+     const/4 v2, 0x0
+ 
+     iput v2, v1, Landroid/widget/Editor;->mCursorCount:I
+ 
+     return-void
+ .end method
+ 
  .method protected setCursorPosition_internal(II)V
      .locals 1
      .parameter "start"
***************
*** 23804,23809 ****
--- 23847,23862 ----
      return-void
  .end method
  
+ .method public final setSoftInputShownOnFocus(Z)V
+     .locals 0
+     .parameter "show"
+ 
+     .prologue
+     invoke-virtual {p0, p1}, Landroid/widget/TextView;->setShowSoftInputOnFocus(Z)V
+ 
+     return-void
+ .end method
+ 
  .method protected setSpan_internal(Ljava/lang/Object;III)V
      .locals 1
      .parameter "span"
***************
*** 24539,24544 ****
--- 24592,24605 ----
      .parameter "tf"
  
      .prologue
+     invoke-virtual {p0}, Landroid/widget/TextView;->getContext()Landroid/content/Context;
+ 
+     move-result-object v0
+ 
+     invoke-static {v0, p1}, Lmiui/util/UiUtils;->replaceTypeface(Landroid/content/Context;Landroid/graphics/Typeface;)Landroid/graphics/Typeface;
+ 
+     move-result-object p1
+ 
      iget-object v0, p0, Landroid/widget/TextView;->mTextPaint:Landroid/text/TextPaint;
  
      invoke-virtual {v0}, Landroid/text/TextPaint;->getTypeface()Landroid/graphics/Typeface;
***************
*** 24586,24591 ****
--- 24647,24656 ----
      :goto_0
      invoke-virtual {p0, p1}, Landroid/widget/TextView;->setTypeface(Landroid/graphics/Typeface;)V
  
+     invoke-virtual {p0}, Landroid/widget/TextView;->getTypeface()Landroid/graphics/Typeface;
+ 
+     move-result-object p1
+ 
      if-eqz p1, :cond_2
  
      invoke-virtual {p1}, Landroid/graphics/Typeface;->getStyle()I
***************
*** 24656,24661 ****
--- 24721,24730 ----
  
      invoke-virtual {v2, v3}, Landroid/text/TextPaint;->setTextSkewX(F)V
  
+     invoke-static {p1}, Landroid/graphics/MiuiTypeface;->getDefaultTypeface(Landroid/graphics/Typeface;)Landroid/graphics/Typeface;
+ 
+     move-result-object p1
+ 
      invoke-virtual {p0, p1}, Landroid/widget/TextView;->setTypeface(Landroid/graphics/Typeface;)V
  
      goto :goto_3
