*** ./smali/android/widget/Toast$TN.smali	2013-11-10 11:52:50.369671942 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/android/widget/Toast$TN.smali	2013-11-10 11:54:11.145674011 +0800
***************
*** 172,177 ****
--- 172,186 ----
  
  
  # virtual methods
+ .method getParams()Landroid/view/WindowManager$LayoutParams;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/Toast$TN;->mParams:Landroid/view/WindowManager$LayoutParams;
+ 
+     return-object v0
+ .end method
+ 
  .method public handleHide()V
      .locals 2
  
