*** ./smali/com/android/server/wm/WindowStateAnimator.smali	2013-11-10 11:52:23.353671251 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/services.jar.out/./smali/com/android/server/wm/WindowStateAnimator.smali	2013-11-10 11:53:42.793673285 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/wm/WindowStateAnimator$SurfaceTrace;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/wm/WindowStateAnimator$SurfaceTrace;,
!         Lcom/android/server/wm/WindowStateAnimator$Injector;
      }
  .end annotation
  
***************
*** 110,115 ****
--- 111,118 ----
  
  .field mLocalAnimating:Z
  
+ .field mMiuiHidden:Z
+ 
  .field mPendingDestroySurface:Landroid/view/Surface;
  
  .field final mPolicy:Landroid/view/WindowManagerPolicy;
***************
*** 5070,5078 ****
  
      iput-boolean v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mSurfaceShown:Z
  
!     iget-object v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mSurface:Landroid/view/Surface;
! 
!     invoke-virtual {v3}, Landroid/view/Surface;->show()V
  
      iget-object v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mWin:Lcom/android/server/wm/WindowState;
  
--- 5073,5079 ----
  
      iput-boolean v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mSurfaceShown:Z
  
!     invoke-static {p0}, Lcom/android/server/wm/WindowStateAnimator$Injector;->showSurface(Lcom/android/server/wm/WindowStateAnimator;)V
  
      iget-object v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mWin:Lcom/android/server/wm/WindowState;
  
