*** ./smali/com/android/server/pm/Installer.smali	2013-11-08 15:41:03.380921508 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/services.jar.out/./smali/com/android/server/pm/Installer.smali	2013-11-08 15:38:08.056921529 +0800
***************
*** 1,4 ****
! .class public final Lcom/android/server/pm/Installer;
  .super Ljava/lang/Object;
  .source "Installer.java"
  
--- 1,4 ----
! .class public Lcom/android/server/pm/Installer;
  .super Ljava/lang/Object;
  .source "Installer.java"
  
***************
*** 586,591 ****
--- 586,603 ----
  
  
  # virtual methods
+ .method public callExecute(Ljava/lang/String;)I
+     .locals 1
+     .parameter "cmd"
+ 
+     .prologue
+     invoke-direct {p0, p1}, Lcom/android/server/pm/Installer;->execute(Ljava/lang/String;)I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public clearUserData(Ljava/lang/String;I)I
      .locals 3
      .parameter "name"
