*** ./smali/com/android/server/pm/UserManagerService.smali	2013-11-10 11:52:25.049671294 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/services.jar.out/./smali/com/android/server/pm/UserManagerService.smali	2013-11-10 11:53:44.485673328 +0800
***************
*** 3050,3056 ****
--- 3050,3065 ----
  
      monitor-enter v1
  
+     if-nez p1, :cond_0
+ 
+     const/4 v0, 0x1
+ 
      :try_start_0
+     monitor-exit v1
+ 
+     return v0
+ 
+     :cond_0
      iget-object v0, p0, Lcom/android/server/pm/UserManagerService;->mUserIds:[I
  
      invoke-static {v0, p1}, Lcom/android/internal/util/ArrayUtils;->contains([II)Z
