*** ./smali/android/database/CursorWindow.smali	2013-11-18 12:21:38.058342687 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/database/CursorWindow.smali	2013-11-18 12:18:41.926339380 +0800
***************
*** 6,11 ****
--- 6,19 ----
  .implements Landroid/os/Parcelable;
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Landroid/database/CursorWindow$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field public static final CREATOR:Landroid/os/Parcelable$Creator; = null
      .annotation system Ldalvik/annotation/Signature;
***************
*** 635,640 ****
--- 643,652 ----
      :cond_0
      sget-object v0, Landroid/database/CursorWindow;->sWindowToPidMap:Landroid/util/SparseIntArray;
  
+     invoke-static {v0, p1}, Landroid/database/CursorWindow$Injector;->delQuota(Landroid/util/SparseIntArray;I)V
+ 
+     sget-object v0, Landroid/database/CursorWindow;->sWindowToPidMap:Landroid/util/SparseIntArray;
+ 
      invoke-virtual {v0, p1}, Landroid/util/SparseIntArray;->delete(I)V
  
      monitor-exit v1
***************
*** 703,708 ****
--- 715,722 ----
      invoke-static {v0, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
      :cond_0
+     invoke-static {p1}, Landroid/database/CursorWindow$Injector;->addQuota(I)V
+ 
      monitor-exit v1
  
      return-void
