*** ./smali/android/content/res/Configuration.smali	2013-11-08 15:41:19.200921506 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/content/res/Configuration.smali	2013-11-08 15:38:24.388921527 +0800
***************
*** 154,159 ****
--- 154,167 ----
  
  .field public static final UI_MODE_TYPE_NORMAL:I = 0x1
  
+ .field public static final UI_MODE_TYPE_SCALE_HUGE:I = 0xf
+ 
+ .field public static final UI_MODE_TYPE_SCALE_LARGE:I = 0xe
+ 
+ .field public static final UI_MODE_TYPE_SCALE_MEDIUM:I = 0xd
+ 
+ .field public static final UI_MODE_TYPE_SCALE_SMALL:I = 0xc
+ 
  .field public static final UI_MODE_TYPE_TELEVISION:I = 0x4
  
  .field public static final UI_MODE_TYPE_UNDEFINED:I
***************
*** 168,173 ****
--- 176,183 ----
  
  .field public densityDpi:I
  
+ .field public extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
  .field public fontScale:F
  
  .field public hardKeyboardHidden:I
***************
*** 226,260 ****
  .end method
  
  .method public constructor <init>()V
!     .locals 0
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 0
      .parameter "o"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 0
      .parameter "source"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
--- 236,288 ----
  .end method
  
  .method public constructor <init>()V
!     .locals 1
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 1
      .parameter "o"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 1
      .parameter "source"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
***************
*** 293,306 ****
  
      and-int/2addr v0, p0
  
!     if-eqz v0, :cond_0
  
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
--- 321,341 ----
  
      and-int/2addr v0, p0
  
!     if-nez v0, :cond_0
! 
!     invoke-static {p0}, Lmiui/content/res/ExtraConfiguration;->needNewResources(I)Z
! 
!     move-result v0
  
+     if-eqz v0, :cond_1
+ 
+     :cond_0
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_1
      const/4 v0, 0x0
  
      goto :goto_0
***************
*** 680,685 ****
--- 715,728 ----
  
      sub-int v2, v3, v4
  
+     iget-object v3, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v4, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v3, v4}, Lmiui/content/res/ExtraConfiguration;->compareTo(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v2
+ 
      goto/16 :goto_0
  .end method
  
***************
*** 967,972 ****
--- 1010,1025 ----
      or-int/lit16 v0, v0, 0x1000
  
      :cond_11
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v2, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1, v2}, Lmiui/content/res/ExtraConfiguration;->diff(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v1
+ 
+     or-int/2addr v0, v1
+ 
      return v0
  .end method
  
***************
*** 1162,1168 ****
  
      add-int v0, v1, v2
  
!     return v0
  
      :cond_0
      const/4 v1, 0x0
--- 1215,1229 ----
  
      add-int v0, v1, v2
  
!     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
! 
!     invoke-virtual {v1}, Lmiui/content/res/ExtraConfiguration;->hashCode()I
! 
!     move-result v1
! 
!     add-int/2addr v1, v0
! 
!     return v1
  
      :cond_0
      const/4 v1, 0x0
***************
*** 1417,1422 ****
--- 1478,1487 ----
  
      iput v0, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, p1}, Lmiui/content/res/ExtraConfiguration;->readFromParcel(Landroid/os/Parcel;)V
+ 
      return-void
  
      :cond_1
***************
*** 1469,1475 ****
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 1
      .parameter "o"
  
      .prologue
--- 1534,1540 ----
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 2
      .parameter "o"
  
      .prologue
***************
*** 1572,1577 ****
--- 1637,1648 ----
  
      iput v0, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v1, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, v1}, Lmiui/content/res/ExtraConfiguration;->setTo(Lmiui/content/res/ExtraConfiguration;)V
+ 
      return-void
  .end method
  
***************
*** 1589,1594 ****
--- 1660,1667 ----
  
      iput v1, p0, Landroid/content/res/Configuration;->mcc:I
  
+     invoke-direct {p0}, Landroid/content/res/Configuration;->loadPreOperator()V
+ 
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
***************
*** 1629,1634 ****
--- 1702,1710 ----
  
      iput v1, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1}, Lmiui/content/res/ExtraConfiguration;->setToDefaults()V
      return-void
  .end method
  
***************
*** 1953,1958 ****
--- 2029,2042 ----
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
      :cond_0
+     iget-object v2, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v2}, Lmiui/content/res/ExtraConfiguration;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
      const/16 v2, 0x7d
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
***************
*** 2845,2850 ****
--- 2929,2944 ----
      iput v1, p0, Landroid/content/res/Configuration;->seq:I
  
      :cond_18
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v2, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1, v2}, Lmiui/content/res/ExtraConfiguration;->updateFrom(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v1
+ 
+     or-int/2addr v0, v1
+ 
      return v0
  
      :cond_19
***************
*** 2964,2969 ****
--- 3058,3067 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, p1, p2}, Lmiui/content/res/ExtraConfiguration;->writeToParcel(Landroid/os/Parcel;I)V
+ 
      return-void
  
      :cond_0
***************
*** 3000,3002 ****
--- 3098,3149 ----
  
      goto :goto_1
  .end method
+ 
+ .method private loadPreOperator()V
+     .locals 3
+ 
+     .prologue
+     const/16 v2, 0x1cc
+ 
+     const-string v1, "ro.carrier.name"
+ 
+     invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     .local v0, carrier:Ljava/lang/String;
+     const-string v1, "ct"
+ 
+     invoke-virtual {v1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_1
+ 
+     iput v2, p0, Landroid/content/res/Configuration;->mcc:I
+ 
+     const/4 v1, 0x3
+ 
+     iput v1, p0, Landroid/content/res/Configuration;->mnc:I
+ 
+     :cond_0
+     :goto_0
+     return-void
+ 
+     :cond_1
+     const-string v1, "cu"
+ 
+     invoke-virtual {v1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     iput v2, p0, Landroid/content/res/Configuration;->mcc:I
+ 
+     const/4 v1, 0x1
+ 
+     iput v1, p0, Landroid/content/res/Configuration;->mnc:I
+ 
+     goto :goto_0
+ .end method
\ No newline at end of file
