*** ./smali/android/telephony/PhoneNumberUtils.smali	2013-11-18 12:21:32.722342587 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/telephony/PhoneNumberUtils.smali	2013-11-18 12:18:36.650339280 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/telephony/PhoneNumberUtils$CountryCallingCodeAndNewIndex;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/telephony/PhoneNumberUtils$CountryCallingCodeAndNewIndex;,
!         Landroid/telephony/PhoneNumberUtils$Injector;
      }
  .end annotation
  
***************
*** 827,832 ****
--- 828,845 ----
      .end packed-switch
  .end method
  
+ .method static callIndexOfLastNetworkChar(Ljava/lang/String;)I
+     .locals 1
+     .parameter "str"
+ 
+     .prologue
+     invoke-static {p0}, Landroid/telephony/PhoneNumberUtils;->indexOfLastNetworkChar(Ljava/lang/String;)I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public static calledPartyBCDFragmentToString([BII)Ljava/lang/String;
      .locals 2
      .parameter "bytes"
***************
*** 1670,1683 ****
  
      if-ge v6, v10, :cond_b
  
!     invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v10
  
      sub-int v2, v10, v7
  
      .local v2, effectiveALen:I
!     invoke-virtual {p1}, Ljava/lang/String;->length()I
  
      move-result v10
  
--- 1683,1696 ----
  
      if-ge v6, v10, :cond_b
  
!     invoke-static {p0}, Landroid/telephony/PhoneNumberUtils$Injector;->getEffectiveLength(Ljava/lang/String;)I
  
      move-result v10
  
      sub-int v2, v10, v7
  
      .local v2, effectiveALen:I
!     invoke-static {p1}, Landroid/telephony/PhoneNumberUtils$Injector;->getEffectiveLength(Ljava/lang/String;)I
  
      move-result v10
  
***************
*** 2024,2030 ****
      :cond_c
      const/16 v18, 0x0
  
!     move-object/from16 v0, p1
  
      move/from16 v1, v18
  
--- 2037,2043 ----
      :cond_c
      const/16 v18, 0x0
  
!     move-object/from16 v0, p0
  
      move/from16 v1, v18
  
***************
*** 2124,2130 ****
      :cond_14
      move-object/from16 v0, p1
  
!     invoke-static {v0, v10, v4}, Landroid/telephony/PhoneNumberUtils;->checkPrefixIsIgnorable(Ljava/lang/String;II)Z
  
      move-result v18
  
--- 2137,2143 ----
      :cond_14
      move-object/from16 v0, p1
  
!     invoke-static {v0, v11, v4}, Landroid/telephony/PhoneNumberUtils;->checkPrefixIsIgnorable(Ljava/lang/String;II)Z
  
      move-result v18
  
***************
*** 3105,3110 ****
--- 3118,3127 ----
      .parameter "defaultCountryIso"
  
      .prologue
+     invoke-static {p0}, Lmiui/telephony/PhoneNumberUtils;->removeDashesAndBlanks(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object p0
+ 
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v1
***************
*** 4102,4107 ****
--- 4119,4139 ----
  
      move-result-object p0
  
+     invoke-static {p0}, Lmiui/telephony/PhoneNumberUtils;->parseNumber(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object p0
+ 
+     invoke-static {p0, p2}, Lmiui/telephony/PhoneNumberUtils;->isMiuiEmergencyNumber(Ljava/lang/String;Z)Z
+ 
+     move-result v8
+ 
+     if-eqz v8, :cond_miui_0
+ 
+     move v6, v7
+ 
+     goto :goto_0
+ 
+     :cond_miui_0
      const-string v8, "ril.ecclist"
  
      invoke-static {v8}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
***************
*** 5483,5488 ****
--- 5515,5522 ----
      move-result v0
  
      .local v0, c:C
+     invoke-static {v4, v0, v2}, Landroid/telephony/PhoneNumberUtils$Injector;->appendNonSeparator(Ljava/lang/StringBuilder;CI)V
+ 
      const/16 v5, 0xa
  
      invoke-static {v0, v5}, Ljava/lang/Character;->digit(CI)I
