*** ./smali/android/telephony/TelephonyManager.smali	2013-11-18 12:21:32.630342585 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/telephony/TelephonyManager.smali	2013-11-18 12:18:36.558339280 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "TelephonyManager.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Landroid/telephony/TelephonyManager$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field public static final ACTION_PHONE_STATE_CHANGED:Ljava/lang/String; = "android.intent.action.PHONE_STATE"
  
***************
*** 248,253 ****
--- 256,270 ----
      return-object v0
  .end method
  
+ .method static getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     sget-object v0, Landroid/telephony/TelephonyManager;->sContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public static getDefault()Landroid/telephony/TelephonyManager;
      .locals 1
  
***************
*** 1851,1856 ****
--- 1868,1877 ----
  
      move-result-object v0
  
+     invoke-static {v0}, Landroid/telephony/TelephonyManager$Injector;->getSimOperator(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
      return-object v0
  .end method
  
