*** ./smali/com/android/internal/app/LocalePicker.smali	2013-11-08 15:41:48.148921504 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/com/android/internal/app/LocalePicker.smali	2013-11-08 15:38:55.392921525 +0800
***************
*** 359,364 ****
--- 359,366 ----
      :cond_4
      invoke-static {v6}, Ljava/util/Arrays;->sort([Ljava/lang/Object;)V
  
+     invoke-static {v6}, Lmiui/util/ExtraLocalePicker;->adjustLocaleOrder([Lcom/android/internal/app/LocalePicker$LocaleInfo;)V
+ 
      const-string v2, "layout_inflater"
  
      move-object/from16 v0, p0
