*** ./smali/com/android/internal/app/LocalePicker.smali	2013-11-10 11:53:32.157673012 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/com/android/internal/app/LocalePicker.smali	2013-11-10 11:54:52.005675057 +0800
***************
*** 359,364 ****
--- 359,366 ----
      :cond_4
      invoke-static {v6}, Ljava/util/Arrays;->sort([Ljava/lang/Object;)V
  
+     invoke-static {v6}, Lmiui/util/ExtraLocalePicker;->adjustLocaleOrder([Lcom/android/internal/app/LocalePicker$LocaleInfo;)V
+ 
      const-string v2, "layout_inflater"
  
      move-object/from16 v0, p0
