*** ./smali/com/android/internal/app/ResolverActivity.smali	2013-11-10 11:53:32.297673016 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/com/android/internal/app/ResolverActivity.smali	2013-11-10 11:54:52.141675060 +0800
***************
*** 11,17 ****
      value = {
          Lcom/android/internal/app/ResolverActivity$ItemLongClickListener;,
          Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;,
!         Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;
      }
  .end annotation
  
--- 11,18 ----
      value = {
          Lcom/android/internal/app/ResolverActivity$ItemLongClickListener;,
          Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;,
!         Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;,
!         Lcom/android/internal/app/ResolverActivity$Injector;
      }
  .end annotation
  
***************
*** 371,377 ****
  
      .prologue
      .local p5, rList:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
!     const v1, 0x103030f
  
      invoke-virtual {p0, v1}, Lcom/android/internal/app/ResolverActivity;->setTheme(I)V
  
--- 372,378 ----
  
      .prologue
      .local p5, rList:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
!     const v1, 0x60d003e
  
      invoke-virtual {p0, v1}, Lcom/android/internal/app/ResolverActivity;->setTheme(I)V
  
***************
*** 405,417 ****
  
      iput-boolean v0, p0, Lcom/android/internal/app/ResolverActivity;->mAlwaysUseOption:Z
  
!     invoke-virtual {p0}, Lcom/android/internal/app/ResolverActivity;->getResources()Landroid/content/res/Resources;
! 
!     move-result-object v1
! 
!     const v2, 0x10e0037
! 
!     invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getInteger(I)I
  
      move-result v1
  
--- 406,412 ----
  
      iput-boolean v0, p0, Lcom/android/internal/app/ResolverActivity;->mAlwaysUseOption:Z
  
!     invoke-static {p0}, Lcom/android/internal/app/ResolverActivity$Injector;->getMaxColumns(Lcom/android/internal/app/ResolverActivity;)I
  
      move-result v1
  
***************
*** 505,510 ****
--- 500,514 ----
  
      :cond_1
      :goto_1
+     iget-object v0, v9, Lcom/android/internal/app/AlertController$AlertParams;->mView:Landroid/view/View;
+ 
+     if-eqz v0, :cond_miui_00
+ 
+     move/from16 v0, p6
+ 
+     invoke-static {p0, v0}, Lcom/android/internal/app/ResolverActivity$Injector;->initialize(Lcom/android/internal/app/ResolverActivity;Z)V
+ 
+     :cond_miui_00
      return-void
  
      .end local v8           #am:Landroid/app/ActivityManager;
***************
*** 588,593 ****
--- 592,602 ----
      invoke-virtual {p0}, Lcom/android/internal/app/ResolverActivity;->setupAlert()V
  
      if-eqz p6, :cond_1
+     invoke-static {p0}, Lmiui/util/UiUtils;->isV5Ui(Landroid/content/Context;)Z
+ 
+     move-result v1
+ 
+     if-nez v1, :cond_1
  
      const v1, 0x1020278
  
***************
*** 834,846 ****
  
      invoke-virtual {v11, v0}, Landroid/content/IntentFilter;->addDataScheme(Ljava/lang/String;)V
  
!     move-object/from16 v0, p1
! 
!     iget-object v0, v0, Landroid/content/pm/ResolveInfo;->filter:Landroid/content/IntentFilter;
! 
!     move-object/from16 v21, v0
! 
!     invoke-virtual/range {v21 .. v21}, Landroid/content/IntentFilter;->authoritiesIterator()Ljava/util/Iterator;
  
      move-result-object v5
  
--- 843,849 ----
  
      invoke-virtual {v11, v0}, Landroid/content/IntentFilter;->addDataScheme(Ljava/lang/String;)V
  
!     invoke-static/range {p1 .. p1}, Lcom/android/internal/app/ResolverActivity$Injector;->getAuthoritiesIterator(Landroid/content/pm/ResolveInfo;)Ljava/util/Iterator;
  
      move-result-object v5
  
***************
*** 892,904 ****
      .end local v4           #a:Landroid/content/IntentFilter$AuthorityEntry;
      .end local v18           #port:I
      :cond_5
!     move-object/from16 v0, p1
! 
!     iget-object v0, v0, Landroid/content/pm/ResolveInfo;->filter:Landroid/content/IntentFilter;
! 
!     move-object/from16 v21, v0
! 
!     invoke-virtual/range {v21 .. v21}, Landroid/content/IntentFilter;->pathsIterator()Ljava/util/Iterator;
  
      move-result-object v16
  
--- 895,901 ----
      .end local v4           #a:Landroid/content/IntentFilter$AuthorityEntry;
      .end local v18           #port:I
      :cond_5
!     invoke-static/range {p1 .. p1}, Lcom/android/internal/app/ResolverActivity$Injector;->getPathsIterator(Landroid/content/pm/ResolveInfo;)Ljava/util/Iterator;
  
      move-result-object v16
  
***************
*** 1368,1373 ****
--- 1365,1380 ----
  
      return-void
  .end method
+ 
+ .method setAlwaysUseOption(Z)V
+     .locals 0
+     .parameter "alwaysUseOption"
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/app/ResolverActivity;->mAlwaysUseOption:Z
+ 
+     return-void
+ .end method
  
  .method showAppDetails(Landroid/content/pm/ResolveInfo;)V
      .locals 5
