*** ./smali/android/speech/tts/TextToSpeech$Connection.smali	2013-11-18 12:22:00.970343117 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/speech/tts/TextToSpeech$Connection.smali	2013-11-18 12:19:04.302339801 +0800
***************
*** 124,129 ****
--- 124,156 ----
      return-object v0
  .end method
  
+ .method public isConnected()Z
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/speech/tts/TextToSpeech$Connection;->mService:Landroid/speech/tts/ITextToSpeechService;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Landroid/speech/tts/TextToSpeech$Connection;->mService:Landroid/speech/tts/ITextToSpeechService;
+ 
+     invoke-interface {v0}, Landroid/speech/tts/ITextToSpeechService;->asBinder()Landroid/os/IBinder;
+ 
+     move-result-object v0
+ 
+     invoke-interface {v0}, Landroid/os/IBinder;->isBinderAlive()Z
+ 
+     move-result v0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public onServiceConnected(Landroid/content/ComponentName;Landroid/os/IBinder;)V
      .locals 5
      .parameter "name"
