*** ./smali/android/net/Proxy.smali	2013-11-18 12:22:11.966343324 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/net/Proxy.smali	2013-11-18 12:19:14.886339999 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/net/Proxy$AndroidProxySelectorRoutePlanner;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/net/Proxy$AndroidProxySelectorRoutePlanner;,
!         Landroid/net/Proxy$Injector;
      }
  .end annotation
  
***************
*** 287,293 ****
      .local v1, sa:Ljava/net/InetSocketAddress;
      new-instance v2, Lorg/apache/http/HttpHost;
  
!     invoke-virtual {v1}, Ljava/net/InetSocketAddress;->getHostName()Ljava/lang/String;
  
      move-result-object v3
  
--- 288,294 ----
      .local v1, sa:Ljava/net/InetSocketAddress;
      new-instance v2, Lorg/apache/http/HttpHost;
  
!     invoke-static {v1}, Landroid/net/Proxy$Injector;->getHostName(Ljava/net/InetSocketAddress;)Ljava/lang/String;
  
      move-result-object v3
  
