*** ./smali/android/os/IVibratorService$Stub.smali	2013-11-18 12:21:43.410342788 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/os/IVibratorService$Stub.smali	2013-11-18 12:18:47.198339478 +0800
***************
*** 32,38 ****
  
  .field static final TRANSACTION_vibrate:I = 0x2
  
! .field static final TRANSACTION_vibratePattern:I = 0x3
  
  
  # direct methods
--- 32,40 ----
  
  .field static final TRANSACTION_vibrate:I = 0x2
  
! .field static final TRANSACTION_vibrateEx:I = 0x5
! 
! .field static final TRANSACTION_vibratePattern:I = 0x4
  
  
  # direct methods
***************
*** 222,234 ****
--- 224,257 ----
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
+     
+    :sswitch_5
+     const-string v5, "android.os.IVibratorService"
+ 
+     invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     invoke-virtual {p2}, Landroid/os/Parcel;->createByteArray()[B
+ 
+     move-result-object v0
+ 
+     .local v0, _arg0:[B
+     invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
+ 
+     move-result-object v2
+ 
+     .restart local v2       #_arg1:Landroid/os/IBinder;
+     invoke-virtual {p0, v0, v2}, Landroid/os/IVibratorService$Stub;->vibrateEx([BLandroid/os/IBinder;)V
+ 
+     invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     goto :goto_0
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
          0x2 -> :sswitch_2
          0x3 -> :sswitch_3
          0x4 -> :sswitch_4
+         0x5 -> :sswitch_5
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
