*** ./smali/android/os/IVibratorService$Stub$Proxy.smali	2013-11-18 12:21:43.698342793 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/os/IVibratorService$Stub$Proxy.smali	2013-11-18 12:18:47.482339485 +0800
***************
*** 237,242 ****
--- 237,300 ----
      throw v2
  .end method
  
+ .method public vibrateEx([BLandroid/os/IBinder;)V
+     .locals 5
+     .parameter "params"
+     .parameter "token"
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v0
+ 
+     .local v0, _data:Landroid/os/Parcel;
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v1
+ 
+     .local v1, _reply:Landroid/os/Parcel;
+     :try_start_0
+     const-string v2, "android.os.IVibratorService"
+ 
+     invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
+ 
+     invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeByteArray([B)V
+ 
+     invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
+ 
+     iget-object v2, p0, Landroid/os/IVibratorService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
+ 
+     const/4 v3, 0x5
+ 
+     const/4 v4, 0x0
+ 
+     invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     return-void
+ 
+     :catchall_0
+     move-exception v2
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     throw v2
+ .end method
+ 
  .method public vibratePattern([JILandroid/os/IBinder;)V
      .locals 5
      .parameter "pattern"
