*** ./smali/android/os/Process.smali	2013-11-10 11:53:22.445672764 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/android/os/Process.smali	2013-11-10 11:54:42.001674801 +0800
***************
*** 26,31 ****
--- 26,33 ----
  
  .field public static final FIRST_SHARED_APPLICATION_GID:I = 0xc350
  
+ .field public static final FM_RADIO_UID:I = 0x405
+ 
  .field public static final GOOGLE_SHARED_APP_CONTENT:Ljava/lang/String; = "com.google.process.content"
  
  .field public static final LAST_APPLICATION_UID:I = 0x4e1f
