*** ./smali/com/android/server/AppWidgetService.smali	2013-11-10 11:52:22.925671240 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/services.jar.out/./smali/com/android/server/AppWidgetService.smali	2013-11-10 11:53:42.377673274 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "AppWidgetService.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Lcom/android/server/AppWidgetService$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field private static final TAG:Ljava/lang/String; = "AppWidgetService"
  
***************
*** 1190,1195 ****
--- 1198,1205 ----
  
      invoke-virtual {v0, p1}, Lcom/android/server/AppWidgetServiceImpl;->systemReady(Z)V
  
+     invoke-static {p0}, Lcom/android/server/AppWidgetService$Injector;->receiveRestoreFinish(Lcom/android/server/AppWidgetService;)V
+ 
      iget-object v0, p0, Lcom/android/server/AppWidgetService;->mContext:Landroid/content/Context;
  
      iget-object v1, p0, Lcom/android/server/AppWidgetService;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
