*** ./smali/com/android/server/AppWidgetServiceImpl.smali	2013-11-18 12:21:31.010342555 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/services.jar.out/./smali/com/android/server/AppWidgetServiceImpl.smali	2013-11-18 12:18:34.962339250 +0800
***************
*** 9,15 ****
          Lcom/android/server/AppWidgetServiceImpl$ServiceConnectionProxy;,
          Lcom/android/server/AppWidgetServiceImpl$AppWidgetId;,
          Lcom/android/server/AppWidgetServiceImpl$Host;,
!         Lcom/android/server/AppWidgetServiceImpl$Provider;
      }
  .end annotation
  
--- 9,16 ----
          Lcom/android/server/AppWidgetServiceImpl$ServiceConnectionProxy;,
          Lcom/android/server/AppWidgetServiceImpl$AppWidgetId;,
          Lcom/android/server/AppWidgetServiceImpl$Host;,
!         Lcom/android/server/AppWidgetServiceImpl$Provider;,
!         Lcom/android/server/AppWidgetServiceImpl$Injector;
      }
  .end annotation
  
***************
*** 6096,6101 ****
--- 6097,6116 ----
  
      move-object/from16 v0, p0
  
+     move/from16 v1, v31
+ 
+     invoke-static {v0, v1}, Lcom/android/server/AppWidgetServiceImpl$Injector;->isDuplicateWidgetId(Lcom/android/server/AppWidgetServiceImpl;I)Z
+ 
+     move-result v31
+ 
+     if-nez v31, :cond_3
+ 
+     iget v0, v12, Lcom/android/server/AppWidgetServiceImpl$AppWidgetId;->appWidgetId:I
+ 
+     move/from16 v31, v0
+ 
+     move-object/from16 v0, p0
+ 
      iget v0, v0, Lcom/android/server/AppWidgetServiceImpl;->mNextAppWidgetId:I
  
      move/from16 v32, v0
***************
*** 6728,6733 ****
--- 6743,6792 ----
  
      throw v0
  .end method
+ 
+ .method reload()V
+     .locals 2
+ 
+     .prologue
+     iget-object v1, p0, Lcom/android/server/AppWidgetServiceImpl;->mAppWidgetIds:Ljava/util/ArrayList;
+ 
+     monitor-enter v1
+ 
+     :try_start_0
+     iget-object v0, p0, Lcom/android/server/AppWidgetServiceImpl;->mAppWidgetIds:Ljava/util/ArrayList;
+ 
+     invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
+ 
+     iget-object v0, p0, Lcom/android/server/AppWidgetServiceImpl;->mHosts:Ljava/util/ArrayList;
+ 
+     invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
+ 
+     iget-object v0, p0, Lcom/android/server/AppWidgetServiceImpl;->mInstalledProviders:Ljava/util/ArrayList;
+ 
+     invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
+ 
+     const/4 v0, 0x0
+ 
+     iput-boolean v0, p0, Lcom/android/server/AppWidgetServiceImpl;->mStateLoaded:Z
+ 
+     monitor-exit v1
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     invoke-virtual {p0}, Lcom/android/server/AppWidgetServiceImpl;->sendInitialBroadcasts()V
+ 
+     return-void
+ 
+     :catchall_0
+     move-exception v0
+ 
+     :try_start_1
+     monitor-exit v1
+     :try_end_1
+     .catchall {:try_start_1 .. :try_end_1} :catchall_0
+ 
+     throw v0
+ .end method
  
  .method removeProviderLocked(ILcom/android/server/AppWidgetServiceImpl$Provider;)V
      .locals 5
