*** ./smali/com/android/server/BackupManagerService.smali	2013-11-10 11:52:17.193671093 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/services.jar.out/./smali/com/android/server/BackupManagerService.smali	2013-11-10 11:53:36.449673122 +0800
***************
*** 31,37 ****
          Lcom/android/server/BackupManagerService$RestoreParams;,
          Lcom/android/server/BackupManagerService$RestoreGetSetsParams;,
          Lcom/android/server/BackupManagerService$ProvisionedObserver;,
!         Lcom/android/server/BackupManagerService$BackupRequest;
      }
  .end annotation
  
--- 31,38 ----
          Lcom/android/server/BackupManagerService$RestoreParams;,
          Lcom/android/server/BackupManagerService$RestoreGetSetsParams;,
          Lcom/android/server/BackupManagerService$ProvisionedObserver;,
!         Lcom/android/server/BackupManagerService$BackupRequest;,
!         Lcom/android/server/BackupManagerService$Injector;
      }
  .end annotation
  
***************
*** 6997,7003 ****
      :try_start_4
      const-string v2, "fullback"
  
!     invoke-virtual {p0, v12, v2}, Lcom/android/server/BackupManagerService;->startConfirmationUi(ILjava/lang/String;)Z
  
      move-result v2
  
--- 6998,7004 ----
      :try_start_4
      const-string v2, "fullback"
  
!     invoke-static {p0, v12, v2}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationUi(Lcom/android/server/BackupManagerService;ILjava/lang/String;)Z
  
      move-result v2
  
***************
*** 7077,7083 ****
  
      invoke-virtual {v2, v3, v4, v5}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-virtual {p0, v12, v1}, Lcom/android/server/BackupManagerService;->startConfirmationTimeout(ILcom/android/server/BackupManagerService$FullParams;)V
  
      invoke-virtual {p0, v1}, Lcom/android/server/BackupManagerService;->waitForCompletion(Lcom/android/server/BackupManagerService$FullParams;)V
      :try_end_9
--- 7078,7084 ----
  
      invoke-virtual {v2, v3, v4, v5}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-static {p0, v12, v1}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationTimeout(Lcom/android/server/BackupManagerService;ILcom/android/server/BackupManagerService$FullParams;)V
  
      invoke-virtual {p0, v1}, Lcom/android/server/BackupManagerService;->waitForCompletion(Lcom/android/server/BackupManagerService$FullParams;)V
      :try_end_9
***************
*** 7224,7230 ****
      :try_start_4
      const-string v6, "fullrest"
  
!     invoke-virtual {p0, v5, v6}, Lcom/android/server/BackupManagerService;->startConfirmationUi(ILjava/lang/String;)Z
  
      move-result v6
  
--- 7225,7231 ----
      :try_start_4
      const-string v6, "fullrest"
  
!     invoke-static {p0, v5, v6}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationUi(Lcom/android/server/BackupManagerService;ILjava/lang/String;)Z
  
      move-result v6
  
***************
*** 7304,7310 ****
  
      invoke-virtual {v6, v7, v8, v9}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-virtual {p0, v5, v4}, Lcom/android/server/BackupManagerService;->startConfirmationTimeout(ILcom/android/server/BackupManagerService$FullParams;)V
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService;->waitForCompletion(Lcom/android/server/BackupManagerService$FullParams;)V
      :try_end_9
--- 7305,7311 ----
  
      invoke-virtual {v6, v7, v8, v9}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-static {p0, v5, v4}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationTimeout(Lcom/android/server/BackupManagerService;ILcom/android/server/BackupManagerService$FullParams;)V
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService;->waitForCompletion(Lcom/android/server/BackupManagerService$FullParams;)V
      :try_end_9
***************
*** 7584,7589 ****
--- 7585,7599 ----
      throw v2
  .end method
  
+ .method getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/BackupManagerService;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public getCurrentTransport()Ljava/lang/String;
      .locals 3
  
