*** ./smali/com/android/server/ConnectivityService$DefaultNetworkFactory.smali	2013-11-18 12:21:25.882342458 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/services.jar.out/./smali/com/android/server/ConnectivityService$DefaultNetworkFactory.smali	2013-11-18 12:18:29.974339156 +0800
***************
*** 132,137 ****
--- 132,146 ----
  
      goto :goto_0
  
+     :pswitch_7
+     new-instance v0, Landroid/net/UsbShareNetStateTracker;
+ 
+     iget-object v1, p2, Landroid/net/NetworkConfig;->name:Ljava/lang/String;
+ 
+     invoke-direct {v0, p1, v1}, Landroid/net/UsbShareNetStateTracker;-><init>(ILjava/lang/String;)V
+ 
+     goto :goto_0
+ 
      nop
  
      :pswitch_data_0
***************
*** 146,150 ****
--- 155,164 ----
          :pswitch_4
          :pswitch_3
          :pswitch_6
+         :pswitch_0
+         :pswitch_0
+         :pswitch_0
+         :pswitch_0
+         :pswitch_7
      .end packed-switch
  .end method
