*** ./smali/com/android/server/InputMethodManagerService.smali	2013-11-10 11:52:22.825671237 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/services.jar.out/./smali/com/android/server/InputMethodManagerService.smali	2013-11-10 11:53:42.277673271 +0800
***************
*** 4341,4346 ****
--- 4341,4348 ----
  
      invoke-virtual {v11, v0}, Landroid/widget/Switch;->setOnCheckedChangeListener(Landroid/widget/CompoundButton$OnCheckedChangeListener;)V
  
+     invoke-virtual/range {p0 .. p0}, Lcom/android/server/InputMethodManagerService;->removeCustomTitle()V
+ 
      new-instance v6, Lcom/android/server/InputMethodManagerService$ImeSubtypeListAdapter;
  
      const v22, 0x10900ca
***************
*** 8031,8036 ****
--- 8033,8068 ----
      throw v0
  .end method
  
+ .method removeCustomTitle()V
+     .locals 3
+ 
+     .prologue
+     const/4 v2, 0x0
+ 
+     iget-object v0, p0, Lcom/android/server/InputMethodManagerService;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v0}, Lmiui/util/UiUtils;->isV5Ui(Landroid/content/Context;)Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/server/InputMethodManagerService;->mDialogBuilder:Landroid/app/AlertDialog$Builder;
+ 
+     invoke-virtual {v0, v2}, Landroid/app/AlertDialog$Builder;->setCustomTitle(Landroid/view/View;)Landroid/app/AlertDialog$Builder;
+ 
+     iget-object v0, p0, Lcom/android/server/InputMethodManagerService;->mDialogBuilder:Landroid/app/AlertDialog$Builder;
+ 
+     const v1, 0x1040453
+ 
+     invoke-virtual {v0, v1}, Landroid/app/AlertDialog$Builder;->setTitle(I)Landroid/app/AlertDialog$Builder;
+ 
+     iput-object v2, p0, Lcom/android/server/InputMethodManagerService;->mSwitchingDialogTitleView:Landroid/view/View;
+ 
+     :cond_0
+     return-void
+ .end method
+ 
  .method public setAdditionalInputMethodSubtypes(Ljava/lang/String;[Landroid/view/inputmethod/InputMethodSubtype;)V
      .locals 10
      .parameter "imiId"
