*** ./smali/com/android/server/UiModeManagerService.smali	2013-11-18 12:21:27.842342495 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/services.jar.out/./smali/com/android/server/UiModeManagerService.smali	2013-11-18 12:18:31.930339193 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "UiModeManagerService.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Lcom/android/server/UiModeManagerService$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field private static final ENABLE_LAUNCH_CAR_DOCK_APP:Z = true
  
***************
*** 48,53 ****
--- 56,63 ----
  
  .field private mNightMode:I
  
+ .field mNormalType:I
+ 
  .field private mNotificationManager:Landroid/app/NotificationManager;
  
  .field private final mPowerManager:Landroid/os/PowerManager;
***************
*** 97,102 ****
--- 107,114 ----
  
      invoke-direct {p0}, Landroid/app/IUiModeManager$Stub;-><init>()V
  
+     iput v1, p0, Lcom/android/server/UiModeManagerService;->mNormalType:I
+ 
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0}, Landroid/os/Handler;-><init>()V
***************
*** 187,192 ****
--- 199,208 ----
  
      invoke-virtual {v0, v3, v4}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
+ 
+     invoke-static {p0, v0}, Lcom/android/server/UiModeManagerService$Injector;->registerUIModeScaleChangedObserver(Lcom/android/server/UiModeManagerService;Landroid/content/Context;)V
+ 
      const-string v0, "power"
  
      invoke-virtual {p1, v0}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 948,954 ****
  
      .end local v0           #uiMode:I
      :cond_2
!     iget v0, p0, Lcom/android/server/UiModeManagerService;->mDefaultUiModeType:I
  
      goto :goto_0
  
--- 964,970 ----
  
      .end local v0           #uiMode:I
      :cond_2
!     iget v0, p0, Lcom/android/server/UiModeManagerService;->mNormalType:I
  
      goto :goto_0
  
