*** ./smali/com/android/server/VibratorService.smali	2013-11-10 11:52:17.373671097 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/services.jar.out/./smali/com/android/server/VibratorService.smali	2013-11-10 11:53:36.633673127 +0800
***************
*** 1112,1117 ****
--- 1112,1126 ----
      throw v1
  .end method
  
+ .method public vibrateEx([BLandroid/os/IBinder;)V
+     .locals 0
+     .parameter "params"
+     .parameter "token"
+ 
+     .prologue
+     return-void
+ .end method
+ 
  .method public vibratePattern([JILandroid/os/IBinder;)V
      .locals 9
      .parameter "pattern"
