*** ./smali/com/android/server/location/GpsLocationProvider.smali	2013-11-08 15:41:02.612921508 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/services.jar.out/./smali/com/android/server/location/GpsLocationProvider.smali	2013-11-08 15:38:07.288921529 +0800
***************
*** 12,18 ****
          Lcom/android/server/location/GpsLocationProvider$NetworkLocationListener;,
          Lcom/android/server/location/GpsLocationProvider$ProviderHandler;,
          Lcom/android/server/location/GpsLocationProvider$Listener;,
!         Lcom/android/server/location/GpsLocationProvider$GpsRequest;
      }
  .end annotation
  
--- 12,19 ----
          Lcom/android/server/location/GpsLocationProvider$NetworkLocationListener;,
          Lcom/android/server/location/GpsLocationProvider$ProviderHandler;,
          Lcom/android/server/location/GpsLocationProvider$Listener;,
!         Lcom/android/server/location/GpsLocationProvider$GpsRequest;,
!         Lcom/android/server/location/GpsLocationProvider$Injector;
      }
  .end annotation
  
***************
*** 3477,3482 ****
--- 3478,3485 ----
  
      invoke-virtual {v2, v6, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     invoke-static {p0, v2}, Lcom/android/server/location/GpsLocationProvider$Injector;->appendUidExtra(Lcom/android/server/location/GpsLocationProvider;Landroid/content/Intent;)V
+ 
      iget-object v6, p0, Lcom/android/server/location/GpsLocationProvider;->mContext:Landroid/content/Context;
  
      sget-object v8, Landroid/os/UserHandle;->ALL:Landroid/os/UserHandle;
***************
*** 4949,4954 ****
--- 4952,4966 ----
      return-void
  .end method
  
+ .method getClientUids()[I
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/location/GpsLocationProvider;->mClientUids:[I
+ 
+     return-object v0
+ .end method
+ 
  .method public getGpsStatusProvider()Landroid/location/IGpsStatusProvider;
      .locals 1
  
***************
*** 4967,4972 ****
--- 4979,4993 ----
      return-object v0
  .end method
  
+ .method getNavigating()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/server/location/GpsLocationProvider;->mNavigating:Z
+ 
+     return v0
+ .end method
+ 
  .method public getNetInitiatedListener()Landroid/location/INetInitiatedListener;
      .locals 1
  
