*** ./smali/android/webkit/WebViewClassic.smali	2013-11-10 11:53:21.637672743 +0800
--- /opt/OpenSouce/MonkeyMiCode/9508cm.new/temp/new_smali/framework.jar.out/./smali/android/webkit/WebViewClassic.smali	2013-11-10 11:54:41.097674778 +0800
***************
*** 16155,16160 ****
--- 16155,16169 ----
      goto :goto_0
  .end method
  
+ .method public isHitSelectionHandle()Z
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
  .method public isPaused()Z
      .locals 1
  
***************
*** 16307,16312 ****
--- 16316,16330 ----
      goto :goto_0
  .end method
  
+ .method public isSelectingText()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/webkit/WebViewClassic;->mSelectingText:Z
+ 
+     return v0
+ .end method
+ 
  .method public loadData(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V
      .locals 0
      .parameter "data"
***************
*** 21326,21332 ****
      goto :goto_0
  .end method
  
! .method sendOurVisibleRect()Landroid/graphics/Rect;
      .locals 4
  
      .prologue
--- 21344,21350 ----
      goto :goto_0
  .end method
  
! .method public sendOurVisibleRect()Landroid/graphics/Rect;
      .locals 4
  
      .prologue
***************
*** 22493,22498 ****
--- 22511,22547 ----
      return-void
  .end method
  
+ .method public setProxy(Ljava/lang/String;I)V
+     .locals 3
+     .parameter "host"
+     .parameter "port"
+ 
+     .prologue
+     const/4 v2, 0x0
+ 
+     const/16 v1, 0xc1
+ 
+     if-eqz p1, :cond_0
+ 
+     if-nez p2, :cond_1
+ 
+     :cond_0
+     invoke-static {v1, v2}, Landroid/webkit/WebViewCore;->sendStaticMessage(ILjava/lang/Object;)V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_1
+     new-instance v0, Landroid/net/ProxyProperties;
+ 
+     invoke-direct {v0, p1, p2, v2}, Landroid/net/ProxyProperties;-><init>(Ljava/lang/String;ILjava/lang/String;)V
+ 
+     .local v0, pp:Landroid/net/ProxyProperties;
+     invoke-static {v1, v0}, Landroid/webkit/WebViewCore;->sendStaticMessage(ILjava/lang/Object;)V
+ 
+     goto :goto_0
+ .end method
+ 
  .method public setScrollBarStyle(I)V
      .locals 1
      .parameter "style"
***************
*** 23359,23361 ****
--- 23408,23417 ----
  
      return v0
  .end method
+ 
+ .method public hideFloatView()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
