*** ./smali/android/net/wifi/WifiConfiguration$1.smali	2013-11-08 15:41:42.668921505 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiConfiguration$1.smali	2013-11-08 15:38:50.000921524 +0800
***************
*** 85,90 ****
--- 85,114 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
+     invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v6
+ 
+     iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->wapiASCert:Ljava/lang/String;
+ 
+     invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v6
+ 
+     iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->wapiUserCert:Ljava/lang/String;
+ 
+     invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v6
+ 
+     iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->wapiPsk:Ljava/lang/String;
+ 
+     invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v6
+ 
+     iput v6, v1, Landroid/net/wifi/WifiConfiguration;->wapiPskType:I
+ 
      const/4 v3, 0x0
  
      .local v3, i:I
