*** ./smali/android/text/format/DateFormat.smali	2013-11-08 15:41:09.684921507 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/android/text/format/DateFormat.smali	2013-11-08 15:38:14.532921528 +0800
***************
*** 3,8 ****
--- 3,16 ----
  .source "DateFormat.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Landroid/text/format/DateFormat$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field public static final AM_PM:C = 'a'
  
***************
*** 1269,1274 ****
--- 1277,1286 ----
      move-result-object v4
  
      .local v4, value:Ljava/lang/String;
+     invoke-static {p0, v4}, Landroid/text/format/DateFormat$Injector;->check24HourFormatForChina(Landroid/content/Context;Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v4
+ 
      if-nez v4, :cond_3
  
      invoke-virtual {p0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
