*** ./smali/com/android/internal/telephony/CallerInfoAsyncQuery.smali	2013-11-18 12:22:12.290343330 +0800
--- /opt/opensource/micode/9508cm/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/CallerInfoAsyncQuery.smali	2013-11-18 12:19:15.206340005 +0800
***************
*** 10,16 ****
          Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;,
          Lcom/android/internal/telephony/CallerInfoAsyncQuery$QueryPoolException;,
          Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;,
!         Lcom/android/internal/telephony/CallerInfoAsyncQuery$OnQueryCompleteListener;
      }
  .end annotation
  
--- 10,17 ----
          Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;,
          Lcom/android/internal/telephony/CallerInfoAsyncQuery$QueryPoolException;,
          Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;,
!         Lcom/android/internal/telephony/CallerInfoAsyncQuery$OnQueryCompleteListener;,
!         Lcom/android/internal/telephony/CallerInfoAsyncQuery$Injector;
      }
  .end annotation
  
***************
*** 313,319 ****
      :cond_0
      sget-object v0, Landroid/provider/ContactsContract$PhoneLookup;->CONTENT_FILTER_URI:Landroid/net/Uri;
  
!     invoke-static {p2}, Landroid/net/Uri;->encode(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
--- 314,324 ----
      :cond_0
      sget-object v0, Landroid/provider/ContactsContract$PhoneLookup;->CONTENT_FILTER_URI:Landroid/net/Uri;
  
!     invoke-static {p2}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$Injector;->getPhoneNumber(Ljava/lang/String;)Ljava/lang/String;
! 
!     move-result-object v1
! 
!     invoke-static {v1}, Landroid/net/Uri;->encode(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
